<!DOCTYPE html>
<html class="client-nojs" lang="en" dir="ltr">

<!-- Mirrored from en.wikipedia.org/wiki/BioJava by HTTrack Website Copier/3.x [XR&CO'2014], Tue, 17 Jan 2017 12:52:42 GMT -->
<!-- Added by HTTrack --><meta http-equiv="content-type" content="text/html;charset=UTF-8" /><!-- /Added by HTTrack -->
<head>
<meta charset="UTF-8"/>
<title>BioJava - Wikipedia</title>
<script>document.documentElement.className = document.documentElement.className.replace( /(^|\s)client-nojs(\s|$)/, "$1client-js$2" );</script>
<script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":false,"wgNamespaceNumber":0,"wgPageName":"BioJava","wgTitle":"BioJava","wgCurRevisionId":755935617,"wgRevisionId":755935617,"wgArticleId":559858,"wgIsArticle":true,"wgIsRedirect":false,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["Articles needing cleanup from July 2016","All pages needing cleanup","Articles containing how-to sections","Official website not in Wikidata","Bioinformatics","Bioinformatics software","Java platform software"],"wgBreakFrames":false,"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRelevantPageName":"BioJava","wgRelevantArticleId":559858,"wgRequestId":"WH36ewpAMFoAAGKLd5gAAADV","wgIsProbablyEditable":true,"wgRestrictionEdit":[],"wgRestrictionMove":[],"wgFlaggedRevsParams":{"tags":{}},"wgStableRevisionId":null,"wgWikiEditorEnabledModules":{"toolbar":true,"dialogs":true,"preview":false,"publish":false},"wgBetaFeaturesFeatures":[],"wgMediaViewerOnClick":true,"wgMediaViewerEnabledByDefault":true,"wgVisualEditor":{"pageLanguageCode":"en","pageLanguageDir":"ltr","usePageImages":true,"usePageDescriptions":true},"wgPreferredVariant":"en","wgMFDisplayWikibaseDescriptions":{"search":true,"nearby":true,"watchlist":true,"tagline":false},"wgRelatedArticles":null,"wgRelatedArticlesUseCirrusSearch":true,"wgRelatedArticlesOnlyUseCirrusSearch":false,"wgULSCurrentAutonym":"English","wgNoticeProject":"wikipedia","wgCentralNoticeCookiesToDelete":[],"wgCentralNoticeCategoriesUsingLegacy":["Fundraising","fundraising"],"wgCategoryTreePageCategoryOptions":"{\"mode\":0,\"hideprefix\":20,\"showcount\":true,\"namespaces\":false}","wgWikibaseItemId":"Q1036510","wgCentralAuthMobileDomain":false,"wgVisualEditorToolbarScrollOffset":0,"wgEditSubmitButtonLabelPublish":false});mw.loader.state({"ext.globalCssJs.user.styles":"ready","ext.globalCssJs.site.styles":"ready","site.styles":"ready","noscript":"ready","user.styles":"ready","user":"ready","user.options":"loading","user.tokens":"loading","ext.cite.styles":"ready","ext.pygments":"ready","wikibase.client.init":"ready","ext.visualEditor.desktopArticleTarget.noscript":"ready","ext.uls.interlanguage":"ready","ext.wikimediaBadges":"ready","mediawiki.legacy.shared":"ready","mediawiki.legacy.commonPrint":"ready","mediawiki.sectionAnchor":"ready","mediawiki.skinning.interface":"ready","skins.vector.styles":"ready","ext.globalCssJs.user":"ready","ext.globalCssJs.site":"ready"});mw.loader.implement("user.options@0j3lz3q",function($,jQuery,require,module){mw.user.options.set({"variant":"en"});});mw.loader.implement("user.tokens@1dqfd7l",function ( $, jQuery, require, module ) {
mw.user.tokens.set({"editToken":"+\\","patrolToken":"+\\","watchToken":"+\\","csrfToken":"+\\"});/*@nomin*/;

});mw.loader.load(["ext.cite.a11y","mediawiki.toc","mediawiki.action.view.postEdit","site","mediawiki.page.startup","mediawiki.user","mediawiki.hidpi","mediawiki.page.ready","mediawiki.legacy.wikibits","mediawiki.searchSuggest","ext.gadget.teahouse","ext.gadget.ReferenceTooltips","ext.gadget.watchlist-notice","ext.gadget.DRN-wizard","ext.gadget.charinsert","ext.gadget.refToolbar","ext.gadget.extra-toolbar-buttons","ext.gadget.switcher","ext.gadget.featured-articles-links","ext.centralauth.centralautologin","mmv.head","mmv.bootstrap.autostart","ext.visualEditor.desktopArticleTarget.init","ext.visualEditor.targetLoader","ext.eventLogging.subscriber","ext.wikimediaEvents","ext.navigationTiming","ext.uls.eventlogger","ext.uls.init","ext.uls.interface","ext.quicksurveys.init","ext.centralNotice.geoIP","ext.centralNotice.startUp","skins.vector.js"]);});</script>
<link rel="stylesheet" href="https://en.wikipedia.org/w/load.php?debug=false&amp;lang=en&amp;modules=ext.cite.styles%7Cext.pygments%2CwikimediaBadges%7Cext.uls.interlanguage%7Cext.visualEditor.desktopArticleTarget.noscript%7Cmediawiki.legacy.commonPrint%2Cshared%7Cmediawiki.sectionAnchor%7Cmediawiki.skinning.interface%7Cskins.vector.styles%7Cwikibase.client.init&amp;only=styles&amp;skin=vector"/>
<script async="" src="../w/load2c8d.php?debug=false&amp;lang=en&amp;modules=startup&amp;only=scripts&amp;skin=vector"></script>
<meta name="ResourceLoaderDynamicStyles" content=""/>
<link rel="stylesheet" href="../w/load18b6.css?debug=false&amp;lang=en&amp;modules=site.styles&amp;only=styles&amp;skin=vector"/>
<meta name="generator" content="MediaWiki 1.29.0-wmf.7"/>
<meta name="referrer" content="origin-when-cross-origin"/>
<link rel="alternate" href="https://en.wikipedia.org/wiki/android-app://org.wikipedia/http/en.m.wikipedia.org/wiki/BioJava"/>
<link rel="alternate" type="application/x-wiki" title="Edit this page" href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit"/>
<link rel="edit" title="Edit this page" href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit"/>
<link rel="apple-touch-icon" href="https://en.wikipedia.org/static/apple-touch/wikipedia.png"/>
<link rel="shortcut icon" href="../static/favicon/wikipedia.ico"/>
<link rel="search" type="application/opensearchdescription+xml" href="../w/opensearch_desc.php" title="Wikipedia (en)"/>
<link rel="EditURI" type="application/rsd+xml" href="../w/api251f.php?action=rsd"/>
<link rel="copyright" href="../../creativecommons.org/licenses/by-sa/3.0/index.html"/>
<link rel="canonical" href="BioJava.html"/>
<link rel="dns-prefetch" href="../../login.wikimedia.org/index.html"/>
<link rel="dns-prefetch" href="../../meta.wikimedia.org/index.html" />
</head>
<body class="mediawiki ltr sitedir-ltr mw-hide-empty-elt ns-0 ns-subject page-BioJava rootpage-BioJava skin-vector action-view">		<div id="mw-page-base" class="noprint"></div>
		<div id="mw-head-base" class="noprint"></div>
		<div id="content" class="mw-body" role="main">
			<a id="top"></a>

							<div id="siteNotice"><!-- CentralNotice --></div>
						<div class="mw-indicators">
</div>
			<h1 id="firstHeading" class="firstHeading" lang="en">BioJava</h1>
									<div id="bodyContent" class="mw-body-content">
									<div id="siteSub">From Wikipedia, the free encyclopedia</div>
								<div id="contentSub"></div>
												<div id="jump-to-nav" class="mw-jump">
					Jump to:					<a href="#mw-head">navigation</a>, 					<a href="#p-search">search</a>
				</div>
				<div id="mw-content-text" lang="en" dir="ltr" class="mw-content-ltr"><script>function mfTempOpenSection(id){var block=document.getElementById("mf-section-"+id);block.className+=" open-block";block.previousSibling.className+=" open-block";}</script><table class="plainlinks metadata ambox ambox-content" role="presentation">
<tr>
<td class="mbox-image">
<div style="width:52px"><img alt="" src="http://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Ambox_important.svg/40px-Ambox_important.svg.png" width="40" height="40" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Ambox_important.svg/60px-Ambox_important.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Ambox_important.svg/80px-Ambox_important.svg.png 2x" data-file-width="40" data-file-height="40" /></div>
</td>
<td class="mbox-text"><span class="mbox-text-span">This article <b>contains <a href="https://en.wikipedia.org/wiki/Wikipedia:What_Wikipedia_is_not#NOTHOWTO" title="Wikipedia:What Wikipedia is not">instructions, advice, or how-to content</a></b>. <span class="hide-when-compact">The purpose of Wikipedia is to present facts, not to train. Please help <a class="external text" href="http://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit">improve this article</a> either by rewriting the how-to content or by <a href="https://meta.wikimedia.org/wiki/Help:Transwiki" class="extiw" title="m:Help:Transwiki">moving</a> it to <a href="https://en.wikiversity.org/wiki/" class="extiw" title="v:">Wikiversity</a>, <a href="https://en.wikibooks.org/wiki/" class="extiw" title="b:">Wikibooks</a> or <a href="https://en.wikivoyage.org/wiki/" class="extiw" title="voy:">Wikivoyage</a>.</span> <small><i>(July 2016)</i></small></span></td>
</tr>
</table>
<table class="infobox vevent" style="width:22em">
<caption class="summary">BioJava</caption>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Software_developer" title="Software developer">Original author(s)</a></th>
<td>Andreas Prlić</td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;">Initial release</th>
<td>2002<span class="noprint">; 15&#160;years ago</span><span style="display:none">&#160;(<span class="bday dtstart published updated">2002</span>)</span></td>
</tr>
<tr>
<td colspan="2" style="text-align:center"></td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Software_release_life_cycle" title="Software release life cycle">Stable release</a></th>
<td>
<div style="margin:0px;">4.2.1 / January&#160;15, 2016<span class="noprint">; 11 months ago</span><span style="display:none">&#160;(<span class="bday dtstart published updated">2016-01-15</span>)</span></div>
</td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Software_release_life_cycle" title="Software release life cycle">Preview release</a></th>
<td>
<div style="margin:0px;">4.2.2 / June&#160;15, 2016<span class="noprint">; 6 months ago</span><span style="display:none">&#160;(<span class="bday dtstart published updated">2016-06-15</span>)</span> nightly builds</div>
</td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Repository_(version_control)" title="Repository (version control)">Repository</a></th>
<td><span class="url"><a rel="nofollow" class="external text" href="http://github.com/biojava">github<wbr />.com<wbr />/biojava</a></span></td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;">Development status</th>
<td>Active</td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;">Written in</th>
<td><a href="Java_(programming_language).html" title="Java (programming language)">Java</a></td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Operating_system" title="Operating system">Operating system</a></th>
<td><a href="https://en.wikipedia.org/wiki/Cross-platform" title="Cross-platform">Cross-platform</a></td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Computing_platform" title="Computing platform">Platform</a></th>
<td><a href="https://en.wikipedia.org/wiki/Web_browser" title="Web browser">Web browser</a> with <a href="Java_(software_platform).html" title="Java (software platform)">Java</a></td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;">Available in</th>
<td>English</td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Software_categories#Broad_categories" title="Software categories">Type</a></th>
<td><a href="https://en.wikipedia.org/wiki/Bioinformatics" title="Bioinformatics">Bioinformatics</a></td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;"><a href="https://en.wikipedia.org/wiki/Software_license" title="Software license">License</a></th>
<td><a href="https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License" title="GNU Lesser General Public License">Lesser GPL</a> 2.1</td>
</tr>
<tr>
<th scope="row" style="white-space: nowrap;">Website</th>
<td><span class="url"><a rel="nofollow" class="external text" href="http://biojava.org/">biojava<wbr />.org</a></span></td>
</tr>
</table>
<p><b>BioJava</b><sup id="cite_ref-BioJava_1-0" class="reference"><a href="#cite_note-BioJava-1">[1]</a></sup> is an <a href="https://en.wikipedia.org/wiki/Open-source_software" title="Open-source software">open-source software</a> project dedicated to provide <a href="Java_(software_platform).html" title="Java (software platform)">Java</a> tools to process <a href="https://en.wikipedia.org/wiki/Biology" title="Biology">biological</a> data.<sup id="cite_ref-pmid18689808_2-0" class="reference"><a href="#cite_note-pmid18689808-2">[2]</a></sup><sup id="cite_ref-Mathura_3-0" class="reference"><a href="#cite_note-Mathura-3">[3]</a></sup> BioJava is a set of <a href="https://en.wikipedia.org/wiki/Library_(computing)" title="Library (computing)">library</a> functions written in the programming language <a href="Java_(programming_language).html" title="Java (programming language)">Java</a> for manipulating sequences, protein structures, file parsers, <a href="https://en.wikipedia.org/wiki/Common_Object_Request_Broker_Architecture" title="Common Object Request Broker Architecture">Common Object Request Broker Architecture</a> (CORBA) interoperability, <a href="https://en.wikipedia.org/wiki/Distributed_Annotation_System" title="Distributed Annotation System">Distributed Annotation System</a> (DAS), access to <a href="https://en.wikipedia.org/wiki/AceDB" title="AceDB">AceDB</a>, dynamic programming, and simple statistical routines. BioJava supports a huge range of data, starting from DNA and protein sequences to the level of 3D protein structures. The BioJava libraries are useful for automating many daily and mundane <a href="https://en.wikipedia.org/wiki/Bioinformatics" title="Bioinformatics">bioinformatics</a> tasks such as to parsing a <a href="https://en.wikipedia.org/wiki/Protein_Data_Bank" title="Protein Data Bank">Protein Data Bank</a> (PDB) file, interacting with Jmol and many more.<sup id="cite_ref-Jmol_4-0" class="reference"><a href="#cite_note-Jmol-4">[4]</a></sup> This <a href="https://en.wikipedia.org/wiki/Application_programming_interface" title="Application programming interface">application programming interface</a> (API) provides various file parsers, data models and algorithms to facilitate working with the standard data formats and enables rapid application development and analysis. These libraries have also been used in developing various extended analysis tools!<sup id="cite_ref-5" class="reference"><a href="#cite_note-5">[5]</a></sup> for example:</p>
<ul>
<li>MUSI: an integrated system to identify multiple specificity from very large peptide or nucleic acid data sets.<sup id="cite_ref-6" class="reference"><a href="#cite_note-6">[6]</a></sup></li>
<li>JEnsembl: a version-aware Java API to Ensembl data systems.<sup id="cite_ref-7" class="reference"><a href="#cite_note-7">[7]</a></sup></li>
<li>Expression profiling of signature gene sets with trinucleotide threading<sup id="cite_ref-8" class="reference"><a href="#cite_note-8">[8]</a></sup></li>
<li>Resolving the structural features of genomic islands: a machine learning approach<sup id="cite_ref-9" class="reference"><a href="#cite_note-9">[9]</a></sup></li>
<li>Utility library for structural bioinformatics<sup id="cite_ref-10" class="reference"><a href="#cite_note-10">[10]</a></sup></li>
</ul>
<p>The BioJava project grew out of work by Thomas Down and Matthew Pocock to create an API to simplify development of Java-based Bioinformatics tools. BioJava is an active open source project that has been developed over more than 12 years and by more than 60 developers. BioJava is one of a number of Bio* projects designed to reduce code duplication.<sup id="cite_ref-pmid12230038_11-0" class="reference"><a href="#cite_note-pmid12230038-11">[11]</a></sup> Examples of such projects that fall under Bio* apart from BioJava are <a href="https://en.wikipedia.org/wiki/BioPython" class="mw-redirect" title="BioPython">BioPython</a>,<sup id="cite_ref-12" class="reference"><a href="#cite_note-12">[12]</a></sup> <a href="https://en.wikipedia.org/wiki/BioPerl" title="BioPerl">BioPerl</a>,<sup id="cite_ref-13" class="reference"><a href="#cite_note-13">[13]</a></sup> <a href="https://en.wikipedia.org/wiki/BioRuby" title="BioRuby">BioRuby</a>,<sup id="cite_ref-14" class="reference"><a href="#cite_note-14">[14]</a></sup> EMBOSS<sup id="cite_ref-15" class="reference"><a href="#cite_note-15">[15]</a></sup> etc.</p>
<p>Version 3.0.5 was a major update to the prior versions. It contained several independent modules. The old project has been moved to a separate project called BioJava-legacy project.<sup id="cite_ref-16" class="reference"><a href="#cite_note-16">[16]</a></sup></p>
<p></p>
<div id="toc" class="toc">
<div id="toctitle">
<h2>Contents</h2>
</div>
<ul>
<li class="toclevel-1 tocsection-1"><a href="#Features"><span class="tocnumber">1</span> <span class="toctext">Features</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="#History_and_publications"><span class="tocnumber">2</span> <span class="toctext">History and publications</span></a></li>
<li class="toclevel-1 tocsection-3"><a href="#Modules"><span class="tocnumber">3</span> <span class="toctext">Modules</span></a>
<ul>
<li class="toclevel-2 tocsection-4"><a href="#Core_Module"><span class="tocnumber">3.1</span> <span class="toctext">Core Module</span></a></li>
<li class="toclevel-2 tocsection-5"><a href="#Protein_structure_modules"><span class="tocnumber">3.2</span> <span class="toctext">Protein structure modules</span></a></li>
<li class="toclevel-2 tocsection-6"><a href="#Genome_and_Sequencing_modules"><span class="tocnumber">3.3</span> <span class="toctext">Genome and Sequencing modules</span></a></li>
<li class="toclevel-2 tocsection-7"><a href="#Alignment_module"><span class="tocnumber">3.4</span> <span class="toctext">Alignment module</span></a></li>
<li class="toclevel-2 tocsection-8"><a href="#ModFinder_module"><span class="tocnumber">3.5</span> <span class="toctext">ModFinder module</span></a></li>
<li class="toclevel-2 tocsection-9"><a href="#Amino_acid_properties_module"><span class="tocnumber">3.6</span> <span class="toctext">Amino acid properties module</span></a></li>
<li class="toclevel-2 tocsection-10"><a href="#Protein_disorder_module"><span class="tocnumber">3.7</span> <span class="toctext">Protein disorder module</span></a>
<ul>
<li class="toclevel-3 tocsection-11"><a href="#Making_library_function_calls"><span class="tocnumber">3.7.1</span> <span class="toctext">Making library function calls</span></a></li>
<li class="toclevel-3 tocsection-12"><a href="#Using_command_line"><span class="tocnumber">3.7.2</span> <span class="toctext">Using command line</span></a></li>
</ul>
</li>
<li class="toclevel-2 tocsection-13"><a href="#Web_service_access_module"><span class="tocnumber">3.8</span> <span class="toctext">Web service access module</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-14"><a href="#Comparisons_with_other_alternatives"><span class="tocnumber">4</span> <span class="toctext">Comparisons with other alternatives</span></a></li>
<li class="toclevel-1 tocsection-15"><a href="#Projects_using_BioJava"><span class="tocnumber">5</span> <span class="toctext">Projects using BioJava</span></a></li>
<li class="toclevel-1 tocsection-16"><a href="#See_also"><span class="tocnumber">6</span> <span class="toctext">See also</span></a></li>
<li class="toclevel-1 tocsection-17"><a href="#External_links"><span class="tocnumber">7</span> <span class="toctext">External links</span></a></li>
<li class="toclevel-1 tocsection-18"><a href="#References"><span class="tocnumber">8</span> <span class="toctext">References</span></a></li>
</ul>
</div>
<p></p>
<h2><span class="mw-headline" id="Features">Features</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=1" title="Edit section: Features">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<p>BioJava provides software modules for many of the typical tasks of bioinformatics programming. These include:</p>
<ul>
<li>Accessing <a href="https://en.wikipedia.org/wiki/Nucleotide_sequence" class="mw-redirect" title="Nucleotide sequence">nucleotide</a> and <a href="https://en.wikipedia.org/wiki/Peptide_sequence" class="mw-redirect" title="Peptide sequence">peptide</a> sequence data from local and remote <a href="https://en.wikipedia.org/wiki/Biological_database" title="Biological database">databases</a></li>
<li>Transforming <a href="https://en.wikipedia.org/wiki/List_of_file_formats#Biology" title="List of file formats">formats</a> of database/ file records</li>
<li>Protein structure parsing and manipulation</li>
<li>Manipulating individual sequences</li>
<li>Searching for similar sequences</li>
<li>Creating and manipulating <a href="https://en.wikipedia.org/wiki/Sequence_alignment" title="Sequence alignment">sequence alignments</a></li>
</ul>
<h2><span class="mw-headline" id="History_and_publications">History and publications</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=2" title="Edit section: History and publications">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<p>In the year 2008, BioJava's first Application note was published.<sup id="cite_ref-pmid18689808_2-1" class="reference"><a href="#cite_note-pmid18689808-2">[2]</a></sup> It was migrated from its original CVS repository to <a href="https://en.wikipedia.org/wiki/Git_hub" class="mw-redirect" title="Git hub">Git hub</a> on April 2013.<sup id="cite_ref-17" class="reference"><a href="#cite_note-17">[17]</a></sup></p>
<p>In October 2012, the most recent paper on BioJava was published.<sup id="cite_ref-BioJava_1-1" class="reference"><a href="#cite_note-BioJava-1">[1]</a></sup> As of November 2012 Google Scholar <a rel="nofollow" class="external text" href="https://scholar.google.com/scholar?cites=3048631375755320177&amp;as_sdt=2005&amp;sciodt=0,5&amp;hl=en">counts more than 130 citations</a>.</p>
<h2><span class="mw-headline" id="Modules">Modules</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=3" title="Edit section: Modules">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<p>During 2014-2015, large parts of the original code base were rewritten. BioJava 3 is a clear departure from the version 1 series. It now consists of several independent modules built using an automation tool called <a href="Apache_Maven.html" title="Apache Maven">Apache Maven</a>.<sup id="cite_ref-18" class="reference"><a href="#cite_note-18">[18]</a></sup> These modules provide state-of-the-art tools for protein structure comparison, pairwise and multiple sequence alignments, working with DNA and protein sequences, analysis of amino acid properties, detecting protein modifications, predicting disordered regions in proteins, and parsers for common file formats using a biologically meaningful data model. The original code has been moved into a separate BioJava legacy project, which is still available for backward compatibility.<sup id="cite_ref-19" class="reference"><a href="#cite_note-19">[19]</a></sup></p>
<p>The following sections will describe several of the new modules and highlight some of the new features that are included in the latest version of BioJava.</p>
<h3><span class="mw-headline" id="Core_Module">Core Module</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=4" title="Edit section: Core Module">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<p>This module provides Java <a href="https://en.wikipedia.org/wiki/Class_(computer_programming)" title="Class (computer programming)">classes</a> to model <a href="https://en.wikipedia.org/wiki/Amino_acid" title="Amino acid">amino acid</a> or <a href="https://en.wikipedia.org/wiki/Nucleotide" title="Nucleotide">nucleotide</a> sequences. The classes were designed so that the names are familiar and make sense to biologists and also provide a concrete representation of the steps in going from a gene sequence to a protein sequence for computer scientists and programmers.</p>
<p>A major change between the legacy BioJava project and BioJava3 lies in the way framework has been designed to exploit then-new innovations in Java. A sequence is defined as a generic <a href="https://en.wikipedia.org/wiki/Interface_(computing)#Software_interfaces_in_object-oriented_languages" title="Interface (computing)">interface</a> allowing the rest of the modules to create any utility that operates on all sequences. Specific classes for common sequences such as DNA and proteins have been defined in order to improve usability for biologists. The translation engine really leverages this work by allowing conversions between DNA, RNA and amino acid sequences. This engine can handle details such as choosing the codon table, converting start codons to methionine, trimming stop codons, specifying the reading frame and handing ambiguous sequences.</p>
<p>Special attentions has been paid to designing the storage of sequences to minimize space needs. Special design patterns such as the <a href="Proxy_pattern.html" title="Proxy pattern">Proxy pattern</a> allowed the developers to create the framework such that sequences can be stored in memory, fetched on demand from a web service such as UniProt, or read from a FASTA file as needed. The latter two approaches save memory by not loading sequence data until it is referenced in the application. This concept can be extended to handle very large genomic datasets, such as NCBI GenBank or a proprietary database.</p>
<h3><span class="mw-headline" id="Protein_structure_modules">Protein structure modules</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=5" title="Edit section: Protein structure modules">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<div class="thumb tright">
<div class="thumbinner" style="width:515px;"><a href="https://en.wikipedia.org/wiki/File:This_window_shows_two_proteins_with_IDs_%224hhb.A%22_and_%224hhb.B%22_aligned_against_each_other.png" class="image"><img alt="" src="http://upload.wikimedia.org/wikipedia/en/b/bd/This_window_shows_two_proteins_with_IDs_%224hhb.A%22_and_%224hhb.B%22_aligned_against_each_other.png" width="513" height="652" class="thumbimage" data-file-width="513" data-file-height="652" /></a>
<div class="thumbcaption">This window shows two proteins with IDs "4hhb.A" and "4hhb.B" aligned against each other. The code is given on the left side. This is produced using BioJava libraries which in turn uses Jmol viewer.<sup id="cite_ref-Jmol_4-1" class="reference"><a href="#cite_note-Jmol-4">[4]</a></sup> The FATCAT<sup id="cite_ref-fatcat_20-0" class="reference"><a href="#cite_note-fatcat-20">[20]</a></sup> rigid algorithm is used here to do the alignment.</div>
</div>
</div>
<p>The protein structure modules provide tools to represent and manipulate 3D biomolecular structures. They focus on protein structure comparison.</p>
<p>The following algorithms have been implemented and included in BioJava.</p>
<ul>
<li>FATCAT algorithm for flexible and rigid body alignment.<sup id="cite_ref-fatcat_20-1" class="reference"><a href="#cite_note-fatcat-20">[20]</a></sup></li>
<li>The standard Combinatorial Extension (CE) algorithm.<sup id="cite_ref-21" class="reference"><a href="#cite_note-21">[21]</a></sup></li>
<li>A new version of CE that can detect circular permutations in proteins.<sup id="cite_ref-22" class="reference"><a href="#cite_note-22">[22]</a></sup></li>
</ul>
<p>These algorithms are used to provide the RCSB Protein Data Bank (PDB)<sup id="cite_ref-23" class="reference"><a href="#cite_note-23">[23]</a></sup> Protein Comparison Tool as well as systematic comparisons of all proteins in the PDB on a weekly basis.<sup id="cite_ref-24" class="reference"><a href="#cite_note-24">[24]</a></sup></p>
<p>Parsers for PDB<sup id="cite_ref-25" class="reference"><a href="#cite_note-25">[25]</a></sup> and mmCIF<sup id="cite_ref-26" class="reference"><a href="#cite_note-26">[26]</a></sup> file formats allow the loading of structure data into a reusable data model. This feature is used by the SIFTS project to map between UniProt sequences and PDB structures.<sup id="cite_ref-27" class="reference"><a href="#cite_note-27">[27]</a></sup> Information from the RCSB PDB can be dynamically fetched without the need to manually download data. For visualization, an interface to the 3D viewer Jmol<sup id="cite_ref-Jmol_4-2" class="reference"><a href="#cite_note-Jmol-4">[4]</a></sup> <a rel="nofollow" class="external free" href="http://www.jmol.org/">http://www.jmol.org/</a> is provided. The team claims that work is underway to improve interaction with the RCSB PDB viewers.<sup id="cite_ref-28" class="reference"><a href="#cite_note-28">[28]</a></sup></p>
<p>Below is an outline of the code to initialize a window that will display and compare two protein sequences. Please bear in mind that this is just an outline of the code. To make this work one will need to import the correct found in the "org.biojava.bio.structure" package and add also handle exceptions by using a try-catch block.</p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">String</span> <span class="n">name1</span> <span class="o">=</span> <span class="s">"4hhb.A"</span><span class="o">;</span>
<span class="n">String</span> <span class="n">name2</span> <span class="o">=</span> <span class="s">"4hhb.B"</span><span class="o">;</span>

<span class="n">AtomCache</span> <span class="n">cache</span> <span class="o">=</span> <span class="k">new</span> <span class="n">AtomCache</span><span class="o">();</span>
        
<span class="n">Structure</span> <span class="n">structure1</span> <span class="o">=</span> <span class="kc">null</span><span class="o">;</span>
<span class="n">Structure</span> <span class="n">structure2</span> <span class="o">=</span> <span class="kc">null</span><span class="o">;</span>

<span class="n">StructureAlignment</span> <span class="n">algorithm</span>  <span class="o">=</span> 
<span class="n">StructureAlignmentFactory</span><span class="o">.</span><span class="na">getAlgorithm</span><span class="o">(</span><span class="n">FatCatRigid</span><span class="o">.</span><span class="na">algorithmName</span><span class="o">);</span>
   
<span class="n">structure1</span> <span class="o">=</span> <span class="n">cache</span><span class="o">.</span><span class="na">getStructure</span><span class="o">(</span><span class="n">name1</span><span class="o">);</span>
<span class="n">structure2</span> <span class="o">=</span> <span class="n">cache</span><span class="o">.</span><span class="na">getStructure</span><span class="o">(</span><span class="n">name2</span><span class="o">);</span>
    
<span class="n">Atom</span><span class="o">[]</span> <span class="n">ca1</span> <span class="o">=</span> <span class="n">StructureTools</span><span class="o">.</span><span class="na">getAtomCAArray</span><span class="o">(</span><span class="n">structure1</span><span class="o">);</span>
<span class="n">Atom</span><span class="o">[]</span> <span class="n">ca2</span> <span class="o">=</span> <span class="n">StructureTools</span><span class="o">.</span><span class="na">getAtomCAArray</span><span class="o">(</span><span class="n">structure2</span><span class="o">);</span>

<span class="n">FatCatParameters</span> <span class="n">params</span> <span class="o">=</span> <span class="k">new</span> <span class="n">FatCatParameters</span><span class="o">();</span>

<span class="n">AFPChain</span> <span class="n">afpChain</span> <span class="o">=</span> <span class="n">algorithm</span><span class="o">.</span><span class="na">align</span><span class="o">(</span><span class="n">ca1</span><span class="o">,</span><span class="n">ca2</span><span class="o">,</span><span class="n">params</span><span class="o">);</span>

<span class="n">afpChain</span><span class="o">.</span><span class="na">setName1</span><span class="o">(</span><span class="n">name1</span><span class="o">);</span>
<span class="n">afpChain</span><span class="o">.</span><span class="na">setName2</span><span class="o">(</span><span class="n">name2</span><span class="o">);</span>

<span class="n">StructureAlignmentDisplay</span><span class="o">.</span><span class="na">display</span><span class="o">(</span><span class="n">afpChain</span><span class="o">,</span> <span class="n">ca1</span><span class="o">,</span> <span class="n">ca2</span><span class="o">);</span>
</pre></div>
<p>The code aligns the two protein sequences "4hhb.A" and "4hhb.B" based on the FATCAT rigid algorithm.<sup id="cite_ref-fatcat_20-2" class="reference"><a href="#cite_note-fatcat-20">[20]</a></sup></p>
<h3><span class="mw-headline" id="Genome_and_Sequencing_modules">Genome and Sequencing modules</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=6" title="Edit section: Genome and Sequencing modules">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<p>This module is focused on the creation of gene sequence objects from the core module. This is realised by supporting the parsing of the following popular standard file formats generated by open source gene prediction applications:</p>
<ul>
<li>GTF files generated by GeneMark<sup id="cite_ref-29" class="reference"><a href="#cite_note-29">[29]</a></sup></li>
<li>GFF2 files generated by GeneID<sup id="cite_ref-30" class="reference"><a href="#cite_note-30">[30]</a></sup></li>
<li>GFF3 files generated by Glimmer<sup id="cite_ref-31" class="reference"><a href="#cite_note-31">[31]</a></sup></li>
</ul>
<p>Then the gene sequence objects are written out as a GFF3 format and is imported into GMOD.<sup id="cite_ref-32" class="reference"><a href="#cite_note-32">[32]</a></sup> These file formats are well defined but what gets written in the file is very flexible.</p>
<p>The following code example takes a 454scaffold file that was used by genemark to predict genes and returns a collection of ChromosomeSequences. Each chromosome sequence maps to a named entry in the fasta file and would contain N gene sequences. The gene sequences can be +/- strand with frame shifts and multiple transcriptions.</p>
<p>Passing the collection of ChromosomeSequences to GeneFeatureHelper.getProteinSequences would return all protein sequences. You can then write the protein sequences to a fasta file.</p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">LinkedHashMap</span><span class="o">&lt;</span><span class="n">String</span><span class="o">,</span> <span class="n">ChromosomeSequence</span><span class="o">&gt;</span> <span class="n">chromosomeSequenceList</span> <span class="o">=</span> 
	<span class="n">GeneFeatureHelper</span><span class="o">.</span><span class="na">loadFastaAddGeneFeaturesFromGeneMarkGTF</span><span class="o">(</span><span class="k">new</span> <span class="n">File</span><span class="o">(</span><span class="s">"454Scaffolds.fna"</span><span class="o">),</span> <span class="k">new</span> <span class="n">File</span><span class="o">(</span><span class="s">"genemark_hmm.gtf"</span><span class="o">));</span>
<span class="n">LinkedHashMap</span><span class="o">&lt;</span><span class="n">String</span><span class="o">,</span> <span class="n">ProteinSequence</span><span class="o">&gt;</span> <span class="n">proteinSequenceList</span> <span class="o">=</span> <span class="n">GeneFeatureHelper</span><span class="o">.</span><span class="na">getProteinSequences</span><span class="o">(</span><span class="n">chromosomeSequenceList</span><span class="o">.</span><span class="na">values</span><span class="o">());</span>
<span class="n">FastaWriterHelper</span><span class="o">.</span><span class="na">writeProteinSequence</span><span class="o">(</span><span class="k">new</span> <span class="n">File</span><span class="o">(</span><span class="s">"genemark_proteins.faa"</span><span class="o">),</span> <span class="n">proteinSequenceList</span><span class="o">.</span><span class="na">values</span><span class="o">());</span>
</pre></div>
<p>You can also output the gene sequence to a fasta file where the coding regions will be upper case and the non-coding regions will be lower case</p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">LinkedHashMap</span><span class="o">&lt;</span><span class="n">String</span><span class="o">,</span> <span class="n">GeneSequence</span><span class="o">&gt;</span> <span class="n">geneSequenceHashMap</span> <span class="o">=</span> <span class="n">GeneFeatureHelper</span><span class="o">.</span><span class="na">getGeneSequences</span><span class="o">(</span><span class="n">chromosomeSequenceList</span><span class="o">.</span><span class="na">values</span><span class="o">());</span>
<span class="n">Collection</span><span class="o">&lt;</span><span class="n">GeneSequence</span><span class="o">&gt;</span> <span class="n">geneSequences</span> <span class="o">=</span> <span class="n">geneSequenceHashMap</span><span class="o">.</span><span class="na">values</span><span class="o">();</span>
<span class="n">FastaWriterHelper</span><span class="o">.</span><span class="na">writeGeneSequence</span><span class="o">(</span><span class="k">new</span> <span class="n">File</span><span class="o">(</span><span class="s">"genemark_genes.fna"</span><span class="o">),</span> <span class="n">geneSequences</span><span class="o">,</span> <span class="kc">true</span><span class="o">);</span>
</pre></div>
<p>You can easily write out a gff3 view of a ChromosomeSequence with the following code.</p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">FileOutputStream</span> <span class="n">fo</span> <span class="o">=</span> <span class="k">new</span> <span class="n">FileOutputStream</span><span class="o">(</span><span class="s">"genemark.gff3"</span><span class="o">);</span>
<span class="n">GFF3Writer</span> <span class="n">gff3Writer</span> <span class="o">=</span> <span class="k">new</span> <span class="n">GFF3Writer</span><span class="o">();</span>
<span class="n">gff3Writer</span><span class="o">.</span><span class="na">write</span><span class="o">(</span><span class="n">fo</span><span class="o">,</span> <span class="n">chromosomeSequenceList</span><span class="o">);</span>
<span class="n">fo</span><span class="o">.</span><span class="na">close</span><span class="o">();</span>
</pre></div>
<p>For providing input-output support for several common variants of the FASTQ file format from the next generation sequencers,<sup id="cite_ref-33" class="reference"><a href="#cite_note-33">[33]</a></sup> a separate sequencing module is provided. It is called the <b>Sequence Module</b> and is contained in the package org.biojava3.sequencing.io.fastq. For samples on how to use this module please go to this <a rel="nofollow" class="external text" href="http://biojava.org/wiki/BioJava:CookBook3:FASTQ">link</a>.</p>
<p>Work is in progress towards providing a complete set of java classes to do conversions between different file formats where the list of supported gene prediction applications and genome browsers will get longer based on end user requests.</p>
<h3><span class="mw-headline" id="Alignment_module">Alignment module</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=7" title="Edit section: Alignment module">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<p>This module contains several classes and methods that allow users to perform pairwise and multiple sequence alignment.</p>
<p><b>Pairwise sequence alignment</b></p>
<p>For optimal global alignment, BioJava implements the <a href="https://en.wikipedia.org/wiki/Needleman–Wunsch_algorithm" title="Needleman–Wunsch algorithm">Needleman-Wunsch</a><sup id="cite_ref-34" class="reference"><a href="#cite_note-34">[34]</a></sup> algorithm and for performing local alignments the <a href="https://en.wikipedia.org/wiki/Smith–Waterman_algorithm" title="Smith–Waterman algorithm">Smith and Waterman's</a><sup id="cite_ref-35" class="reference"><a href="#cite_note-35">[35]</a></sup> algorithm has been implemented. The outputs of both local and global alignments are available in standard formats.</p>
<p>An <b>example</b> on how to use the libraries is shown below.</p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="kd">protected</span> <span class="kt">void</span> <span class="nf">align</span><span class="o">(</span><span class="n">String</span> <span class="n">uniProtID_1</span><span class="o">,</span> <span class="n">String</span> <span class="n">uniProtID_2</span><span class="o">,</span> <span class="n">PairwiseSequenceAlignerType</span> <span class="n">alignmentType</span><span class="o">)</span> <span class="kd">throws</span> <span class="n">IOException</span><span class="o">,</span> <span class="n">Exception</span> <span class="o">{</span>
	<span class="n">ProteinSequence</span> <span class="n">proteinSeq1</span> <span class="o">=</span> <span class="n">FastaReaderHelper</span><span class="o">.</span><span class="na">readFastaProteinSequence</span><span class="o">((</span><span class="k">new</span> <span class="n">URL</span><span class="o">(</span><span class="n">String</span><span class="o">.</span><span class="na">format</span>
		<span class="o">(</span><span class="s">"http://www.uniprot.org/uniprot/%s.fasta"</span><span class="o">,</span> <span class="n">uniProtID_1</span><span class="o">))).</span><span class="na">openStream</span><span class="o">()).</span><span class="na">get</span><span class="o">(</span><span class="n">uniProtID_1</span><span class="o">);</span>
	<span class="n">ProteinSequence</span> <span class="n">proteinSeq2</span> <span class="o">=</span> <span class="n">FastaReaderHelper</span><span class="o">.</span><span class="na">readFastaProteinSequence</span><span class="o">((</span><span class="k">new</span> <span class="n">URL</span><span class="o">(</span><span class="n">String</span><span class="o">.</span><span class="na">format</span>
		<span class="o">(</span><span class="s">"http://www.uniprot.org/uniprot/%s.fasta"</span><span class="o">,</span> <span class="n">uniProtID_2</span><span class="o">))).</span><span class="na">openStream</span><span class="o">()).</span><span class="na">get</span><span class="o">(</span><span class="n">uniProtID_2</span><span class="o">);</span>
	
       	<span class="n">SequencePair</span><span class="o">&lt;</span><span class="n">ProteinSequence</span><span class="o">,</span> <span class="n">AminoAcidCompound</span><span class="o">&gt;</span> <span class="n">result</span> <span class="o">=</span> <span class="n">Alignments</span><span class="o">.</span><span class="na">getPairwiseAlignment</span><span class="o">(</span><span class="n">proteinSeq1</span><span class="o">,</span> <span class="n">proteinSeq2</span><span class="o">,</span>
               	<span class="n">alignmentType</span><span class="o">,</span> <span class="k">new</span> <span class="n">SimpleGapPenalty</span><span class="o">(),</span> <span class="k">new</span> <span class="n">SimpleSubstitutionMatrix</span><span class="o">&lt;</span><span class="n">AminoAcidCompound</span><span class="o">&gt;());</span>
	<span class="n">System</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">println</span><span class="o">(</span><span class="n">result</span><span class="o">.</span><span class="na">toString</span><span class="o">());</span>
<span class="o">}</span>
</pre></div>
<p>An example call to the above function would look something like this:</p>
<p><b>For Global Alignment</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">align</span><span class="o">(</span><span class="s">"Q21691"</span><span class="o">,</span> <span class="s">"Q21495"</span><span class="o">,</span> <span class="n">PairwiseSequenceAlignerType</span><span class="o">.</span><span class="na">GLOBAL</span><span class="o">);</span>
</pre></div>
<p><b>For Local Alignment</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">align</span><span class="o">(</span><span class="s">"Q21691"</span><span class="o">,</span> <span class="s">"Q21495"</span><span class="o">,</span> <span class="n">PairwiseSequenceAlignerType</span><span class="o">.</span><span class="na">LOCAL</span><span class="o">);</span>
</pre></div>
<p>In addition to these two algorithms, there is an implementation of Guan–Uberbacher algorithm<sup id="cite_ref-36" class="reference"><a href="#cite_note-36">[36]</a></sup> which performs global sequence alignment very efficiently since it only uses linear memory.</p>
<p>For <b><a href="https://en.wikipedia.org/wiki/Multiple_Sequence_Alignment" class="mw-redirect" title="Multiple Sequence Alignment">Multiple Sequence Alignment</a></b>, any of the methods discussed above can be used to progressively perform a multiple sequence alignment.</p>
<h3><span class="mw-headline" id="ModFinder_module">ModFinder module</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=8" title="Edit section: ModFinder module">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<div class="thumb tright">
<div class="thumbinner" style="width:488px;"><a href="https://en.wikipedia.org/wiki/File:An_example_application_using_the_ModFinder_module_and_the_protein_structure_module.png" class="image"><img alt="" src="http://upload.wikimedia.org/wikipedia/commons/8/89/An_example_application_using_the_ModFinder_module_and_the_protein_structure_module.png" width="486" height="374" class="thumbimage" data-file-width="486" data-file-height="374" /></a>
<div class="thumbcaption">An example application using the ModFinder module and the protein structure module. Protein modifications are mapped onto the sequence and structure of ferredoxin I (PDB ID 1GAO).<sup id="cite_ref-37" class="reference"><a href="#cite_note-37">[37]</a></sup> Two possible iron–sulfur clusters are shown on the protein sequence (3Fe–4S (F3S): orange triangles/lines; 4Fe–4S (SF4): purple diamonds/ lines). The 4Fe–4S cluster is displayed in the Jmol structure window above the sequence display</div>
</div>
</div>
<p>The ModFinder module provides new methods to identify and classify protein modifications in protein 3D structures. Over 400 different types of protein modifications such as <a href="https://en.wikipedia.org/wiki/Phosphorylation" title="Phosphorylation">phosphorylation</a>, <a href="https://en.wikipedia.org/wiki/Glycosylation" title="Glycosylation">glycosylation</a>, <a href="https://en.wikipedia.org/wiki/Disulfide" title="Disulfide">disulfide bonds metal chelation</a> etc. were collected and curated based on annotations in PSI-MOD,<sup id="cite_ref-38" class="reference"><a href="#cite_note-38">[38]</a></sup> RESID<sup id="cite_ref-39" class="reference"><a href="#cite_note-39">[39]</a></sup> and RCSB PDB.<sup id="cite_ref-40" class="reference"><a href="#cite_note-40">[40]</a></sup> The module also provides an API for detecting protein modifications within protein structures.</p>
<p><b>Example: identify and print all preloaded modifications from a structure<sup id="cite_ref-41" class="reference"><a href="#cite_note-41">[41]</a></sup></b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">Set</span><span class="o">&lt;</span><span class="n">ModifiedCompound</span><span class="o">&gt;</span> <span class="nf">identifyAllModfications</span><span class="o">(</span><span class="n">Structure</span> <span class="n">struc</span><span class="o">)</span> <span class="o">{</span>
	<span class="n">ProteinModificationIdentifier</span> <span class="n">parser</span> <span class="o">=</span> 
		<span class="k">new</span> <span class="n">ProteinModificationIdentifier</span><span class="o">();</span>
	<span class="n">parser</span><span class="o">.</span><span class="na">identify</span><span class="o">(</span><span class="n">struc</span><span class="o">);</span>
	<span class="n">Set</span><span class="o">&lt;</span><span class="n">ModifiedCompound</span><span class="o">&gt;</span> <span class="n">mcs</span> <span class="o">=</span> 
		<span class="n">parser</span><span class="o">.</span><span class="na">getIdentifiedModifiedCompound</span><span class="o">();</span>
	<span class="k">return</span> <span class="n">mcs</span><span class="o">;</span>
<span class="o">}</span>
</pre></div>
<p><b>Example: identify phosphorylation sites in a structure</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">List</span><span class="o">&lt;</span><span class="n">ResidueNumber</span><span class="o">&gt;</span> <span class="nf">identifyPhosphosites</span><span class="o">(</span><span class="n">Structure</span> <span class="n">struc</span><span class="o">)</span> <span class="o">{</span>
	<span class="n">List</span><span class="o">&lt;</span><span class="n">ResidueNumber</span><span class="o">&gt;</span> <span class="n">phosphosites</span> <span class="o">=</span> <span class="k">new</span> <span class="n">ArrayList</span><span class="o">&lt;</span><span class="n">ResidueNumber</span><span class="o">&gt;();</span>
	<span class="n">ProteinModificationIdentifier</span> <span class="n">parser</span> <span class="o">=</span> 
		<span class="k">new</span> <span class="n">ProteinModificationIdentifier</span><span class="o">();</span>
	<span class="n">parser</span><span class="o">.</span><span class="na">identify</span><span class="o">(</span><span class="n">struc</span><span class="o">,</span> 
		<span class="n">ProteinModificationRegistry</span><span class="o">.</span><span class="na">getByKeyword</span><span class="o">(</span><span class="s">"phosphoprotein"</span><span class="o">));</span>
	<span class="n">Set</span><span class="o">&lt;</span><span class="n">ModifiedCompound</span><span class="o">&gt;</span> <span class="n">mcs</span> <span class="o">=</span> <span class="n">parser</span><span class="o">.</span><span class="na">getIdentifiedModifiedCompound</span><span class="o">();</span>
	<span class="k">for</span> <span class="o">(</span><span class="n">ModifiedCompound</span> <span class="n">mc</span> <span class="o">:</span> <span class="n">mcs</span><span class="o">)</span> <span class="o">{</span>
		<span class="n">Set</span><span class="o">&lt;</span><span class="n">StructureGroup</span><span class="o">&gt;</span> <span class="n">groups</span> <span class="o">=</span> <span class="n">mc</span><span class="o">.</span><span class="na">getGroups</span><span class="o">(</span><span class="kc">true</span><span class="o">);</span>
		<span class="k">for</span> <span class="o">(</span><span class="n">StructureGroup</span> <span class="n">group</span> <span class="o">:</span> <span class="n">groups</span><span class="o">)</span> <span class="o">{</span>
			<span class="n">phosphosites</span><span class="o">.</span><span class="na">add</span><span class="o">(</span><span class="n">group</span><span class="o">.</span><span class="na">getPDBResidueNumber</span><span class="o">());</span>
		<span class="o">}</span>
	<span class="o">}</span>
	<span class="k">return</span> <span class="n">phosphosites</span><span class="o">;</span>
<span class="o">}</span>
</pre></div>
<p><b>Demo code to run the above methods</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="kn">import</span> <span class="nn">org.biojava.bio.structure.ResidueNumber</span><span class="o">;</span>
<span class="kn">import</span> <span class="nn">org.biojava.bio.structure.Structure</span><span class="o">;</span>
<span class="kn">import</span> <span class="nn">org.biojava.bio.structure.io.PDBFileReader</span><span class="o">;</span>
<span class="kn">import</span> <span class="nn">org.biojava3.protmod.structure.ProteinModificationIdentifier</span><span class="o">;</span>
 
<span class="kd">public</span> <span class="kd">static</span> <span class="kt">void</span> <span class="nf">main</span><span class="o">(</span><span class="n">String</span><span class="o">[]</span> <span class="n">args</span><span class="o">)</span> <span class="o">{</span>
	<span class="k">try</span> <span class="o">{</span>
		<span class="n">PDBFileReader</span> <span class="n">reader</span> <span class="o">=</span> <span class="k">new</span> <span class="n">PDBFileReader</span><span class="o">();</span>
		<span class="n">reader</span><span class="o">.</span><span class="na">setAutoFetch</span><span class="o">(</span><span class="kc">true</span><span class="o">);</span>
 
		<span class="c1">// identify all modifications from PDB:1CAD and print them</span>
		<span class="n">String</span> <span class="n">pdbId</span> <span class="o">=</span> <span class="s">"1CAD"</span><span class="o">;</span>
		<span class="n">Structure</span> <span class="n">struc</span> <span class="o">=</span> <span class="n">reader</span><span class="o">.</span><span class="na">getStructureById</span><span class="o">(</span><span class="n">pdbId</span><span class="o">);</span>
		<span class="n">Set</span><span class="o">&lt;</span><span class="n">ModifiedCompound</span><span class="o">&gt;</span> <span class="n">mcs</span> <span class="o">=</span> <span class="n">identifyAllModfications</span><span class="o">(</span><span class="n">struc</span><span class="o">);</span>
		<span class="k">for</span> <span class="o">(</span><span class="n">ModifiedCompound</span> <span class="n">mc</span> <span class="o">:</span> <span class="n">mcs</span><span class="o">)</span> <span class="o">{</span>
			<span class="n">System</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">println</span><span class="o">(</span><span class="n">mc</span><span class="o">.</span><span class="na">toString</span><span class="o">());</span>
		<span class="o">}</span>
 
		<span class="c1">// identify all phosphosites from PDB:3MVJ and print them</span>
		<span class="n">pdbId</span> <span class="o">=</span> <span class="s">"3MVJ"</span><span class="o">;</span>
		<span class="n">struc</span> <span class="o">=</span> <span class="n">reader</span><span class="o">.</span><span class="na">getStructureById</span><span class="o">(</span><span class="n">pdbId</span><span class="o">);</span>
		<span class="n">List</span><span class="o">&lt;</span><span class="n">ResidueNumber</span><span class="o">&gt;</span> <span class="n">psites</span> <span class="o">=</span> <span class="n">identifyPhosphosites</span><span class="o">(</span><span class="n">struc</span><span class="o">);</span>
		<span class="k">for</span> <span class="o">(</span><span class="n">ResidueNumber</span> <span class="n">psite</span> <span class="o">:</span> <span class="n">psites</span><span class="o">)</span> <span class="o">{</span>
			<span class="n">System</span><span class="o">.</span><span class="na">out</span><span class="o">.</span><span class="na">println</span><span class="o">(</span><span class="n">psite</span><span class="o">.</span><span class="na">toString</span><span class="o">());</span>
		<span class="o">}</span>
	<span class="o">}</span> <span class="k">catch</span><span class="o">(</span><span class="n">Exception</span> <span class="n">e</span><span class="o">)</span> <span class="o">{</span>
		<span class="n">e</span><span class="o">.</span><span class="na">printStackTrace</span><span class="o">();</span>
	<span class="o">}</span>
<span class="o">}</span>
</pre></div>
<p>There are plans to include further protein modifications by integrating other resources such as <a href="https://en.wikipedia.org/wiki/UniProt" title="UniProt">UniProt</a><sup id="cite_ref-42" class="reference"><a href="#cite_note-42">[42]</a></sup></p>
<h3><span class="mw-headline" id="Amino_acid_properties_module">Amino acid properties module</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=9" title="Edit section: Amino acid properties module">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<p>This module attempts to provide accurate physio-chemical properties of proteins. The properties that can calculated using this module are as follows:</p>
<ul>
<li><a href="https://en.wikipedia.org/wiki/Molecular_mass" title="Molecular mass">Molecular mass</a></li>
<li><a href="https://en.wikipedia.org/wiki/Mass_attenuation_coefficient" title="Mass attenuation coefficient">Extinction coefficient</a></li>
<li><a href="https://en.wikipedia.org/wiki/Instability_index" title="Instability index">Instability index</a></li>
<li>Aliphatic index</li>
<li>Grand average of hydropathy</li>
<li><a href="https://en.wikipedia.org/wiki/Isoelectric_point" title="Isoelectric point">Isoelectric point</a></li>
<li>Amino acid composition</li>
</ul>
<p>The precise molecular weights for common isotopically labelled amino acids are included in this module. There also exists flexibility to define new amino acid molecules with their molecular weights using simple <a href="https://en.wikipedia.org/wiki/XML" title="XML">XML</a> configuration files. This can be useful where the precise mass is of high importance such as <a href="https://en.wikipedia.org/wiki/Mass_spectrometry" title="Mass spectrometry">mass spectrometry</a> experiments.</p>
<h3><span class="mw-headline" id="Protein_disorder_module">Protein disorder module</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=10" title="Edit section: Protein disorder module">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<p>The goal of this module is to provide users ways to find disorders in protein molecules. BioJava includes a Java implementation of the <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/content/21/16/3369.full">RONN</a> predictor. The latest version of BioJava(3.0.5) makes use of Java's support for multithreading to improve performance by up to 3.2 times,<sup id="cite_ref-43" class="reference"><a href="#cite_note-43">[43]</a></sup> on a modern quad-core machine, as compared to the legacy C implementation.</p>
<p>There are two ways to use this module:</p>
<ul>
<li>Using library function calls</li>
<li>Using command line</li>
</ul>
<h4><span class="mw-headline" id="Making_library_function_calls">Making library function calls</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=11" title="Edit section: Making library function calls">edit</a><span class="mw-editsection-bracket">]</span></span></h4>
<p>The following examples show how to use the module and make function calls to get information about protein disorders. The first two examples make library function calls to calculate the probability of disorder for every residue in the sequence provided.</p>
<p>The third and fourth examples demonstrates how easily one can get the disordered regions of the protein.</p>
<p><b>Example 1: Calculate the probability of disorder for every residue in the sequence</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">FastaSequence</span> <span class="n">fsequence</span> <span class="o">=</span> <span class="k">new</span> <span class="n">FastaSequence</span><span class="o">(</span><span class="s">"name"</span><span class="o">,</span> <span class="s">"LLRGRHLMNGTMIMRPWNFLNDHHFPKFFPHLIEQQAIWLADWWRKKHC"</span> <span class="o">+</span>
				<span class="s">"RPLPTRAPTMDQWDHFALIQKHWTANLWFLTFPFNDKWGWIWFLKDWTPGSADQAQRACTWFFCHGHDTN"</span><span class="o">);</span>
<span class="kt">float</span><span class="o">[]</span>	<span class="n">rawProbabilityScores</span> <span class="o">=</span> <span class="n">Jronn</span><span class="o">.</span><span class="na">getDisorderScores</span><span class="o">(</span><span class="n">fsequence</span><span class="o">);</span>
</pre></div>
<p><b>Example 2: Calculate the probability of disorder for every residue in the sequence for all proteins from the FASTA input file</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="kd">final</span> <span class="n">List</span><span class="o">&lt;</span><span class="n">FastaSequence</span><span class="o">&gt;</span> <span class="n">sequences</span> <span class="o">=</span> <span class="n">SequenceUtil</span><span class="o">.</span><span class="na">readFasta</span><span class="o">(</span><span class="k">new</span> <span class="n">FileInputStream</span><span class="o">(</span><span class="s">"src/test/resources/fasta.in"</span><span class="o">));</span>
<span class="n">Map</span><span class="o">&lt;</span><span class="n">FastaSequence</span><span class="o">,</span> <span class="kt">float</span><span class="o">[]&gt;</span> <span class="n">rawProbabilityScores</span> <span class="o">=</span> <span class="n">Jronn</span><span class="o">.</span><span class="na">getDisorderScores</span><span class="o">(</span><span class="n">sequences</span><span class="o">);</span>
</pre></div>
<p><b>Example 3: Get the disordered regions of the protein for a single protein sequence</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="n">FastaSequence</span> <span class="n">fsequence</span> <span class="o">=</span> <span class="k">new</span> <span class="n">FastaSequence</span><span class="o">(</span><span class="s">"Prot1"</span><span class="o">,</span> <span class="s">"LLRGRHLMNGTMIMRPWNFLNDHHFPKFFPHLIEQQAIWLADWWRKKHC"</span> <span class="o">+</span>
				<span class="s">"RPLPTRAPTMDQWDHFALIQKHWTANLWFLTFPFNDKWGWIWFLKDWTPGSADQAQRACTWFFCHGHDTN"</span> <span class="o">+</span>
				<span class="s">"CQIIFEGRNAPERADPMWTGGLNKHIIARGHFFQSNKFHFLERKFCEMAEIERPNFTCRTLDCQKFPWDDP"</span><span class="o">);</span>
<span class="n">Range</span><span class="o">[]</span> <span class="n">ranges</span> <span class="o">=</span> <span class="n">Jronn</span><span class="o">.</span><span class="na">getDisorder</span><span class="o">(</span><span class="n">fsequence</span><span class="o">);</span>
</pre></div>
<p><b>Example 4: Calculate the disordered regions for the proteins from FASTA file</b></p>
<div class="mw-highlight mw-content-ltr" dir="ltr">
<pre>
<span class="kd">final</span> <span class="n">List</span><span class="o">&lt;</span><span class="n">FastaSequence</span><span class="o">&gt;</span> <span class="n">sequences</span> <span class="o">=</span> <span class="n">SequenceUtil</span><span class="o">.</span><span class="na">readFasta</span><span class="o">(</span><span class="k">new</span> <span class="n">FileInputStream</span><span class="o">(</span><span class="s">"src/test/resources/fasta.in"</span><span class="o">));</span>
<span class="n">Map</span><span class="o">&lt;</span><span class="n">FastaSequence</span><span class="o">,</span> <span class="n">Range</span><span class="o">[]&gt;</span> <span class="n">ranges</span> <span class="o">=</span> <span class="n">Jronn</span><span class="o">.</span><span class="na">getDisorder</span><span class="o">(</span><span class="n">sequences</span><span class="o">);</span>
</pre></div>
<h4><span class="mw-headline" id="Using_command_line">Using command line</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=12" title="Edit section: Using command line">edit</a><span class="mw-editsection-bracket">]</span></span></h4>
<p>BioJava module <i>biojava3-protein-disorder</i> can be compiled into a single executable JAR file and run using the following command.<sup id="cite_ref-44" class="reference"><a href="#cite_note-44">[44]</a></sup></p>
<pre>
java -jar &lt;jar_file_name&gt; 
</pre>
<p><b>Options supported by the command line executable</b></p>
<pre>
JRONN version 3.1b usage 1 August 2011:
java -jar JRONN_JAR_NAME -i=inputfile &lt;OPTIONS&gt;

Where -i=input file 
        Input file can contain one or more FASTA formatted sequences.

All OPTIONS are optional

OPTION DETAILED DESCRIPTION:
        -o full path to the output file, if not specified 
        standard out is used

        -d the value of disorder, defaults to 0.5

        -f output format, V for vertical, where the letters 
        of the sequence and corresponding disorder values are 
        output in two column layout. H for horizontal, where the
        disorder values are provided under the letters of the 
        sequence. Letters and values separated by tabulation in
        this case. Defaults to V.

        -s the file name to write execution statistics to.

        -n the number of threads to use. Defaults to the number of 
        cores available on the computer. n=1 mean sequential 
        processing. Valid values are 1 &lt; n &lt; (2 x num_of_cores)
        Default value will give the best performance.
</pre>
<p><b>Examples</b></p>
<p>Predict disorder values for sequences from input file <i>/home/input.fasta</i> output the results to the <i>standard out</i>. Use default disorder value and utilise all <a href="https://en.wikipedia.org/wiki/Central_processing_unit" title="Central processing unit">CPUs</a> available on the computer.</p>
<pre>
java -jar JRONN.JAR -i=/home/input.fasta
</pre>
<p>Predict disorder values for sequences from input file <i>/home/input.fasta</i> output the results in horizontal layout to the <i>/home/jronn.out</i>, collect execution statistics to <i>/home/jronn.stat.txt</i> file and limit the number of threads to two.</p>
<pre>
java -jar JRONN.JAR -i=/home/input.fasta -o=/home/jronn.out -d=0.6 -n=2 -f=H
</pre>
<p>The arguments can be provided in any order.</p>
<h3><span class="mw-headline" id="Web_service_access_module">Web service access module</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=13" title="Edit section: Web service access module">edit</a><span class="mw-editsection-bracket">]</span></span></h3>
<p>As per the current trends in bioinformatics, web based tools are gaining popularity. The web service module allows bioinformatics services to be accessed using <a href="https://en.wikipedia.org/wiki/Representational_state_transfer" title="Representational state transfer">REST</a> protocols. Currently, two services are implemented: NCBI Blast through the Blast URLAPI (previously known as QBlast) and the <a rel="nofollow" class="external text" href="http://hmmer.janelia.org/">HMMER</a> web service.<sup id="cite_ref-45" class="reference"><a href="#cite_note-45">[45]</a></sup></p>
<h2><span class="mw-headline" id="Comparisons_with_other_alternatives">Comparisons with other alternatives</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=14" title="Edit section: Comparisons with other alternatives">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<p>The need for customized software in the field of <a href="https://en.wikipedia.org/wiki/Bioinformatics" title="Bioinformatics">bioinformatics</a> has been addressed by several groups and individuals. Similar to BioJava, <a href="https://en.wikipedia.org/wiki/Open-source_software" title="Open-source software">open-source software</a> projects such as <a href="https://en.wikipedia.org/wiki/BioPerl" title="BioPerl">BioPerl</a>, <a href="https://en.wikipedia.org/wiki/BioPython" class="mw-redirect" title="BioPython">BioPython</a>, and <a href="https://en.wikipedia.org/wiki/BioRuby" title="BioRuby">BioRuby</a> all provide tool-kits with multiple functionality that make it easier to create customized pipelines or analysis.</p>
<p>As the names suggest, the projects mentioned above use different programming languages. All of these APIs offer similar tools so on what criteria should one base their choice? For programmers who are experienced in only one of these languages, the choice is straightforward. However, for a well-rounded bioinformaticist who knows all of these languages and wants to choose the best language for a job, the choice can be made based on the following guidelines given by a software review done on the Bio* tool-kits.<sup id="cite_ref-pmid12230038_11-1" class="reference"><a href="#cite_note-pmid12230038-11">[11]</a></sup></p>
<p>In general, for small programs (&lt;500 lines) that will be used by only an individual or small group, it is hard to beat <a href="https://en.wikipedia.org/wiki/Perl" title="Perl">Perl</a> and <a href="https://en.wikipedia.org/wiki/BioPerl" title="BioPerl">BioPerl</a>. These constraints probably cover the needs of 90 per cent of personal bioinformatics programming.</p>
<p>For beginners, and for writing larger programs in the Bio domain, especially those to be shared and supported by others, <a href="https://en.wikipedia.org/wiki/Python_(programming_language)" title="Python (programming language)">Python’s</a> clarity and brevity make it very attractive.</p>
<p>For those who might be leaning towards a career in bioinformatics and who want to learn only one language, <a href="Java_(programming_language).html" title="Java (programming language)">Java</a> has the widest general programming support, very good support in the Bio domain with BioJava, and is now the de facto language of business (the new COBOL, for better or worse).</p>
<p>Apart from these Bio* projects there is another project called <a rel="nofollow" class="external text" href="https://web.archive.org/web/20130216074401/http://www.charite.de:80/bioinf/strap/Scripting.html#StrapProtein2BiojavaSequence">STRAP</a> which uses Java and aims for similar goals. The STRAP-toolbox, similar to BioJava is also a Java-toolkit for the design of Bioinformatics programs and scripts. The <a rel="nofollow" class="external text" href="http://www.bioinformatics.org/strap/biojava/biojava.html#SIMILARITIES">similarities and differences</a> between BioJava and STRAP are as follows:</p>
<p><b>Similarities</b></p>
<ul>
<li>Both provide comprehensive collections of methods for protein sequences.</li>
<li>Both are used by Java programmers to code bioinformatics algorithms.</li>
<li>Both separate implementations and definitions by using java interfaces.</li>
<li>Both are open source projects.</li>
<li>Both can read and write many sequence file formats.</li>
</ul>
<p><b>Differences</b></p>
<ul>
<li>BioJava is applicable to nucleotide and peptide sequences and can be applied for entire genomes. STRAP cannot cope with single sequences as long as an entire chromosome. Instead STRAP manipulates peptide sequences and 3D- structures of the size of single proteins. Nevertheless, it can hold a high number of sequences and structures in memory. STRAP is designed for protein sequences but can read coding nucleotide files, which are then translated to peptide sequences.</li>
<li>STRAP is very fast since the graphical user interface must be highly responsive. BioJava is used where speed is less critical.</li>
<li>BioJava is well designed in terms of type safety, ontology and object design. BioJava uses objects for sequences, annotations and sequence positions. Even single amino acids or nucleotides are object references. To enhance speed, STRAP avoids frequent object instantiations and invocation of non-final object-methods.
<ul>
<li>In BioJava peptide sequences and nucleotide sequences are lists of symbols. The symbols can be retrieved one after the other with an iterator or sub-sequences can be obtained. The advantages are that the entire sequence does not necessarily reside in memory and that programs are less susceptible to programming errors. <i>Symbol</i> objects are immutable elements of an alphabet. In STRAP however simple byte arrays are used for sequences and float arrays for coordinates. Besides speed the low memory consumption is an important advantage of basic data types. Classes in Strap expose internal data. Therefore, programmers might commit programming errors like manipulating byte arrays directly instead of using the setter methods. Another disadvantage is that no checks are performed in STRAP whether the characters in sequences are valid with respect to an underlying alphabet.</li>
<li>In BioJava sequence positions are realized by the class <i>Location</i>. Discontiguous <i>Location</i> objects are composed of several contiguous <i>RangeLocation</i> objects or <i>PointLocation</i> objects. For the class <i>StrapProtein</i> however, single residue positions are indicated by integer numbers between 0 and <i>countResidues()-1</i>. Multiple positions are given by boolean arrays. True at a given index means selected whereas false means not selected.</li>
</ul>
</li>
<li>BioJava throws exceptions when methods are invoked with invalid parameters. STRAP avoids the time consuming creation of Throwable objects. Instead, errors in methods are indicated by the return values NaN, -1 or null. From the point of program design however <i>Throwable</i> objects are nicer.</li>
<li>In BioJava a <i>Sequence</i> object is either a peptide sequence or a nucleotide sequence. A StrapProtein can hold both at the same time if a coding nucleotide sequence was read and translated into protein. Both, the nucleotide sequence and the peptide sequence are contained in the same StrapProtein object. The coding or non-coding regions can be changed and the peptide sequence alters accordingly.</li>
</ul>
<h2><span class="mw-headline" id="Projects_using_BioJava">Projects using BioJava</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=15" title="Edit section: Projects using BioJava">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<p>The following projects make use of BioJava.</p>
<ul>
<li><a rel="nofollow" class="external text" href="http://www.genostar.com/">Metabolic Pathway Builder</a>: Software suite dedicated to the exploration of connections among genes, proteins, reactions and metabolic pathways</li>
<li><a rel="nofollow" class="external text" href="http://www.dengueinfo.org/">DengueInfo</a>: a Dengue genome information portal that uses BioJava in the middleware and talks to a biosql database.</li>
<li><a rel="nofollow" class="external text" href="http://www.derkholm.net/thomas/dazzle">Dazzle</a>: A BioJava based DAS server.</li>
<li><a rel="nofollow" class="external text" href="http://www.idbs.com/InforSenseSuite/BioSense">BioSense</a>: A <a href="https://en.wikipedia.org/wiki/Plug-in_(computing)" title="Plug-in (computing)">plug-in</a> for the InforSense Suite, an analytics software platform by <a rel="nofollow" class="external text" href="http://www.idbs.com/">IDBS</a> that unitizes BioJava.</li>
<li><a rel="nofollow" class="external text" href="http://www.bioclipse.net/">Bioclipse</a>: A free, open source, workbench for chemo- and bioinformatics with powerful editing and visualizing abilities for molecules, sequences, proteins, spectra, etc.</li>
<li><a rel="nofollow" class="external text" href="http://webclu.bio.wzw.tum.de/prompt">PROMPT</a>: A free, open source framework and application for the comparison and mapping of protein sets. Uses BioJava for handling most input data formats.</li>
<li><a rel="nofollow" class="external text" href="http://www.cytoscape.org/">Cytoscape</a>: An open source bioinformatics software platform to visualize molecular interaction networks.</li>
<li><a rel="nofollow" class="external text" href="http://www.bioweka.org/">BioWeka</a>: An open source biological data mining application.</li>
<li><a rel="nofollow" class="external text" href="http://www.biomatters.com/">Geneious</a>: A molecular biology toolkit.</li>
<li><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/staff/slottad/MassSieve/">MassSieve</a>: An open source application to analyze mass spec proteomics data.</li>
<li><a rel="nofollow" class="external text" href="http://www.bioinformatics.org/strap/">Strap</a>: A tool for multiple sequence alignment and sequence based structure alignment.</li>
<li><a rel="nofollow" class="external text" href="http://www.jstacs.de/">Jstacs</a>: A Java framework for statistical analysis and classification of biological sequences</li>
<li><a rel="nofollow" class="external text" href="http://www.bioinf.jku.at/software/LSTM_protein/">jLSTM</a> "Long Short-Term Memory" for protein classification</li>
<li><a rel="nofollow" class="external text" href="http://lajolla.sourceforge.net/">LaJolla</a> Structural alignment of RNA and proteins using an index structure for fast alignment of thousands of structures. Including an easy to use command line interface. Open source at Sourceforge.</li>
<li><a rel="nofollow" class="external text" href="http://www.geneinfinity.org/genbeans/index.html">GenBeans</a>: A rich client platform for bioinformatics primarily focused on molecular biology and sequence analysis.</li>
</ul>
<h2><span class="mw-headline" id="See_also">See also</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=16" title="Edit section: See also">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<ul>
<li><a href="https://en.wikipedia.org/wiki/Open_Bioinformatics_Foundation" title="Open Bioinformatics Foundation">Open Bioinformatics Foundation</a></li>
<li><a href="https://en.wikipedia.org/wiki/BioPerl" title="BioPerl">BioPerl</a>, <a href="https://en.wikipedia.org/wiki/Biopython" title="Biopython">Biopython</a>, <a href="https://en.wikipedia.org/wiki/BioRuby" title="BioRuby">BioRuby</a></li>
<li><a href="Bioclipse.html" title="Bioclipse">Bioclipse</a></li>
<li><a href="https://en.wikipedia.org/wiki/Comparison_of_software_for_molecular_mechanics_modeling" title="Comparison of software for molecular mechanics modeling">Comparison of software for molecular mechanics modeling</a></li>
</ul>
<h2><span class="mw-headline" id="External_links">External links</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=17" title="Edit section: External links">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<ul>
<li><span class="official-website"><span class="url"><a rel="nofollow" class="external text" href="http://biojava.org/">Official website</a></span></span></li>
<li><a rel="nofollow" class="external text" href="http://www.strubi.ox.ac.uk/RONN/">RONN</a></li>
</ul>
<h2><span class="mw-headline" id="References">References</span><span class="mw-editsection"><span class="mw-editsection-bracket">[</span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit&amp;section=18" title="Edit section: References">edit</a><span class="mw-editsection-bracket">]</span></span></h2>
<div class="reflist columns references-column-count references-column-count-2" style="-moz-column-count: 2; -webkit-column-count: 2; column-count: 2; list-style-type: decimal;">
<ol class="references">
<li id="cite_note-BioJava-1"><span class="mw-cite-backlink">^ <a href="#cite_ref-BioJava_1-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-BioJava_1-1"><sup><i><b>b</b></i></sup></a></span> <span class="reference-text"><cite class="citation journal">Prlić A, Yates A, Bliven SE, et al. (October 2012). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=22877863">"BioJava: an open-source framework for bioinformatics in 2012"</a>. <i>Bioinformatics</i>. <b>28</b> (20): 2693–5. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbts494">10.1093/bioinformatics/bts494</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3467744">3467744</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/22877863">22877863</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=BioJava%3A+an+open-source+framework+for+bioinformatics+in+2012&amp;rft.au=Bliven%2C+SE&amp;rft.aufirst=A&amp;rft.aulast=Prli%C4%87&amp;rft.au=Yates%2C+A&amp;rft.date=2012-10&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3467744&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D22877863&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbts494&amp;rft_id=info%3Apmid%2F22877863&amp;rft.issue=20&amp;rft.jtitle=Bioinformatics&amp;rft.pages=2693-5&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=28" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-pmid18689808-2"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid18689808_2-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid18689808_2-1"><sup><i><b>b</b></i></sup></a></span> <span class="reference-text"><cite class="citation journal">Holland RC, Down TA, Pocock M, Prlić A, Huen D, James K, et al. (2008). <a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2530884">"BioJava: an open-source framework for bioinformatics."</a>. <i>Bioinformatics</i>. <b>24</b> (18): 2096–7. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbtn397">10.1093/bioinformatics/btn397</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2530884">2530884</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/18689808">18689808</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=BioJava%3A+an+open-source+framework+for+bioinformatics.&amp;rft.au=Down%2C+TA&amp;rft.aufirst=RC&amp;rft.au=Huen%2C+D&amp;rft.au=James%2C+K&amp;rft.aulast=Holland&amp;rft.au=Pocock%2C+M&amp;rft.au=Prli%C4%87%2C+A&amp;rft.date=2008&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2530884&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2530884&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbtn397&amp;rft_id=info%3Apmid%2F18689808&amp;rft.issue=18&amp;rft.jtitle=Bioinformatics&amp;rft.pages=2096-7&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=24" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-Mathura-3"><span class="mw-cite-backlink"><b><a href="#cite_ref-Mathura_3-0">^</a></b></span> <span class="reference-text">VS Matha and P Kangueane, 2009, <i>Bioinformatics: a concept-based introduction</i>, 2009. p26</span></li>
<li id="cite_note-Jmol-4"><span class="mw-cite-backlink">^ <a href="#cite_ref-Jmol_4-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-Jmol_4-1"><sup><i><b>b</b></i></sup></a> <a href="#cite_ref-Jmol_4-2"><sup><i><b>c</b></i></sup></a></span> <span class="reference-text">Hanson,R.M. (2010) Jmol a paradigm shift in crystallographic visualization.</span></li>
<li id="cite_note-5"><span class="mw-cite-backlink"><b><a href="#cite_ref-5">^</a></b></span> <span class="reference-text"><a rel="nofollow" class="external autonumber" href="https://scholar.google.com/scholar?cites=3048631375755320177&amp;as_sdt=2005&amp;sciodt=0,5&amp;hl=en">[1]</a></span></li>
<li id="cite_note-6"><span class="mw-cite-backlink"><b><a href="#cite_ref-6">^</a></b></span> <span class="reference-text"><cite class="citation journal">Kim T, Tyndel MS, Huang H, et al. (March 2012). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=22210894">"MUSI: an integrated system for identifying multiple specificity from very large peptide or nucleic acid data sets"</a>. <i>Nucleic Acids Res</i>. <b>40</b> (6): e47. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2Fgkr1294">10.1093/nar/gkr1294</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3315295">3315295</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/22210894">22210894</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=MUSI%3A+an+integrated+system+for+identifying+multiple+specificity+from+very+large+peptide+or+nucleic+acid+data+sets&amp;rft.aufirst=T&amp;rft.au=Huang%2C+H&amp;rft.aulast=Kim&amp;rft.au=Tyndel%2C+MS&amp;rft.date=2012-03&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3315295&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D22210894&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkr1294&amp;rft_id=info%3Apmid%2F22210894&amp;rft.issue=6&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=e47&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=40" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-7"><span class="mw-cite-backlink"><b><a href="#cite_ref-7">^</a></b></span> <span class="reference-text"><cite class="citation journal">Paterson T, Law A (November 2012). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=22945789">"JEnsembl: a version-aware Java API to Ensembl data systems"</a>. <i>Bioinformatics</i>. <b>28</b> (21): 2724–31. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbts525">10.1093/bioinformatics/bts525</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3476335">3476335</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/22945789">22945789</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=JEnsembl%3A+a+version-aware+Java+API+to+Ensembl+data+systems&amp;rft.aufirst=T&amp;rft.aulast=Paterson&amp;rft.au=Law%2C+A&amp;rft.date=2012-11&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3476335&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D22945789&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbts525&amp;rft_id=info%3Apmid%2F22945789&amp;rft.issue=21&amp;rft.jtitle=Bioinformatics&amp;rft.pages=2724-31&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=28" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-8"><span class="mw-cite-backlink"><b><a href="#cite_ref-8">^</a></b></span> <span class="reference-text"><cite class="citation journal">Zajac P, Pettersson E, Gry M, Lundeberg J, Ahmadian A (February 2008). <a rel="nofollow" class="external text" href="http://linkinghub.elsevier.com/retrieve/pii/S0888-7543(07)00254-6">"Expression profiling of signature gene sets with trinucleotide threading"</a>. <i>Genomics</i>. <b>91</b> (2): 209–17. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1016%2Fj.ygeno.2007.10.012">10.1016/j.ygeno.2007.10.012</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/18061398">18061398</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Expression+profiling+of+signature+gene+sets+with+trinucleotide+threading&amp;rft.au=Ahmadian%2C+A&amp;rft.aufirst=P&amp;rft.au=Gry%2C+M&amp;rft.aulast=Zajac&amp;rft.au=Lundeberg%2C+J&amp;rft.au=Pettersson%2C+E&amp;rft.date=2008-02&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0888-7543%2807%2900254-6&amp;rft_id=info%3Adoi%2F10.1016%2Fj.ygeno.2007.10.012&amp;rft_id=info%3Apmid%2F18061398&amp;rft.issue=2&amp;rft.jtitle=Genomics&amp;rft.pages=209-17&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=91" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-9"><span class="mw-cite-backlink"><b><a href="#cite_ref-9">^</a></b></span> <span class="reference-text"><cite class="citation journal">Vernikos GS, Parkhill J (February 2008). <a rel="nofollow" class="external text" href="http://genome.cshlp.org/cgi/pmidlookup?view=long&amp;pmid=18071028">"Resolving the structural features of genomic islands: a machine learning approach"</a>. <i>Genome Res</i>. <b>18</b> (2): 331–42. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1101%2Fgr.7004508">10.1101/gr.7004508</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2203631">2203631</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/18071028">18071028</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Resolving+the+structural+features+of+genomic+islands%3A+a+machine+learning+approach&amp;rft.aufirst=GS&amp;rft.aulast=Vernikos&amp;rft.au=Parkhill%2C+J&amp;rft.date=2008-02&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2203631&amp;rft_id=http%3A%2F%2Fgenome.cshlp.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D18071028&amp;rft_id=info%3Adoi%2F10.1101%2Fgr.7004508&amp;rft_id=info%3Apmid%2F18071028&amp;rft.issue=2&amp;rft.jtitle=Genome+Res.&amp;rft.pages=331-42&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=18" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-10"><span class="mw-cite-backlink"><b><a href="#cite_ref-10">^</a></b></span> <span class="reference-text"><cite class="citation journal">Gront D, Kolinski A (February 2008). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=18227118">"Utility library for structural bioinformatics"</a>. <i>Bioinformatics</i>. <b>24</b> (4): 584–5. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbtm627">10.1093/bioinformatics/btm627</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/18227118">18227118</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Utility+library+for+structural+bioinformatics&amp;rft.aufirst=D&amp;rft.au=Kolinski%2C+A&amp;rft.aulast=Gront&amp;rft.date=2008-02&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D18227118&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbtm627&amp;rft_id=info%3Apmid%2F18227118&amp;rft.issue=4&amp;rft.jtitle=Bioinformatics&amp;rft.pages=584-5&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=24" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-pmid12230038-11"><span class="mw-cite-backlink">^ <a href="#cite_ref-pmid12230038_11-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-pmid12230038_11-1"><sup><i><b>b</b></i></sup></a></span> <span class="reference-text"><cite class="citation journal">Mangalam H (2002). "The Bio* toolkits--a brief overview". <i>Briefings in Bioinformatics</i>. <b>3</b> (3): 296–302. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbib%2F3.3.296">10.1093/bib/3.3.296</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/12230038">12230038</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+Bio%2A+toolkits--a+brief+overview&amp;rft.au=Mangalam+H&amp;rft.date=2002&amp;rft.genre=article&amp;rft_id=info%3Adoi%2F10.1093%2Fbib%2F3.3.296&amp;rft_id=info%3Apmid%2F12230038&amp;rft.issue=3&amp;rft.jtitle=Briefings+in+Bioinformatics&amp;rft.pages=296-302&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=3" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-12"><span class="mw-cite-backlink"><b><a href="#cite_ref-12">^</a></b></span> <span class="reference-text"><cite class="citation journal">Cock PJ, Antao T, Chang JT, et al. (June 2009). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=19304878">"Biopython: freely available Python tools for computational molecular biology and bioinformatics"</a>. <i>Bioinformatics</i>. <b>25</b> (11): 1422–3. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbtp163">10.1093/bioinformatics/btp163</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2682512">2682512</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/19304878">19304878</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Biopython%3A+freely+available+Python+tools+for+computational+molecular+biology+and+bioinformatics&amp;rft.au=Antao%2C+T&amp;rft.au=Chang%2C+JT&amp;rft.aufirst=PJ&amp;rft.aulast=Cock&amp;rft.date=2009-06&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2682512&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D19304878&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbtp163&amp;rft_id=info%3Apmid%2F19304878&amp;rft.issue=11&amp;rft.jtitle=Bioinformatics&amp;rft.pages=1422-3&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=25" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-13"><span class="mw-cite-backlink"><b><a href="#cite_ref-13">^</a></b></span> <span class="reference-text"><cite class="citation journal">Stajich JE, Block D, Boulez K, et al. (October 2002). <a rel="nofollow" class="external text" href="http://genome.cshlp.org/cgi/pmidlookup?view=long&amp;pmid=12368254">"The Bioperl toolkit: Perl modules for the life sciences"</a>. <i>Genome Res</i>. <b>12</b> (10): 1611–8. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1101%2Fgr.361602">10.1101/gr.361602</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC187536">187536</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/12368254">12368254</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+Bioperl+toolkit%3A+Perl+modules+for+the+life+sciences&amp;rft.au=Block%2C+D&amp;rft.au=Boulez%2C+K&amp;rft.aufirst=JE&amp;rft.aulast=Stajich&amp;rft.date=2002-10&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC187536&amp;rft_id=http%3A%2F%2Fgenome.cshlp.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D12368254&amp;rft_id=info%3Adoi%2F10.1101%2Fgr.361602&amp;rft_id=info%3Apmid%2F12368254&amp;rft.issue=10&amp;rft.jtitle=Genome+Res.&amp;rft.pages=1611-8&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=12" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-14"><span class="mw-cite-backlink"><b><a href="#cite_ref-14">^</a></b></span> <span class="reference-text"><cite class="citation journal">Goto N, Prins P, Nakao M, Bonnal R, Aerts J, Katayama T (October 2010). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=20739307">"BioRuby: bioinformatics software for the Ruby programming language"</a>. <i>Bioinformatics</i>. <b>26</b> (20): 2617–9. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbtq475">10.1093/bioinformatics/btq475</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2951089">2951089</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/20739307">20739307</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=BioRuby%3A+bioinformatics+software+for+the+Ruby+programming+language&amp;rft.au=Aerts%2C+J&amp;rft.au=Bonnal%2C+R&amp;rft.aufirst=N&amp;rft.au=Katayama%2C+T&amp;rft.aulast=Goto&amp;rft.au=Nakao%2C+M&amp;rft.au=Prins%2C+P&amp;rft.date=2010-10&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2951089&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D20739307&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbtq475&amp;rft_id=info%3Apmid%2F20739307&amp;rft.issue=20&amp;rft.jtitle=Bioinformatics&amp;rft.pages=2617-9&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=26" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-15"><span class="mw-cite-backlink"><b><a href="#cite_ref-15">^</a></b></span> <span class="reference-text"><cite class="citation journal">Rice P, Longden I, Bleasby A (June 2000). <a rel="nofollow" class="external text" href="http://linkinghub.elsevier.com/retrieve/pii/S0168-9525(00)02024-2">"EMBOSS: the European Molecular Biology Open Software Suite"</a>. <i>Trends Genet</i>. <b>16</b> (6): 276–7. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1016%2FS0168-9525(00)02024-2">10.1016/S0168-9525(00)02024-2</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/10827456">10827456</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=EMBOSS%3A+the+European+Molecular+Biology+Open+Software+Suite&amp;rft.au=Bleasby%2C+A&amp;rft.aufirst=P&amp;rft.aulast=Rice&amp;rft.au=Longden%2C+I&amp;rft.date=2000-06&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0168-9525%2800%2902024-2&amp;rft_id=info%3Adoi%2F10.1016%2FS0168-9525%2800%2902024-2&amp;rft_id=info%3Apmid%2F10827456&amp;rft.issue=6&amp;rft.jtitle=Trends+Genet.&amp;rft.pages=276-7&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=16" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-16"><span class="mw-cite-backlink"><b><a href="#cite_ref-16">^</a></b></span> <span class="reference-text"><a rel="nofollow" class="external text" href="http://www.biojava.org/docs/api1.8.2/">BioJava-legacy</a></span></li>
<li id="cite_note-17"><span class="mw-cite-backlink"><b><a href="#cite_ref-17">^</a></b></span> <span class="reference-text"><cite class="citation web"><a rel="nofollow" class="external text" href="http://biojava.org/wiki/Get_source#History">"History"</a><span class="reference-accessdate">. Retrieved <span class="nowrap">30 Jan</span> 2015</span>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.btitle=History&amp;rft.genre=unknown&amp;rft_id=http%3A%2F%2Fbiojava.org%2Fwiki%2FGet_source%23History&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-18"><span class="mw-cite-backlink"><b><a href="#cite_ref-18">^</a></b></span> <span class="reference-text"><cite class="citation web">Maven, Apache. <a rel="nofollow" class="external text" href="http://maven.apache.org/">"Maven"</a>. Apache.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.aufirst=Apache&amp;rft.aulast=Maven&amp;rft.btitle=Maven&amp;rft.genre=unknown&amp;rft_id=http%3A%2F%2Fmaven.apache.org&amp;rft.pub=Apache&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-19"><span class="mw-cite-backlink"><b><a href="#cite_ref-19">^</a></b></span> <span class="reference-text"><a rel="nofollow" class="external text" href="http://www.biojava.org/docs/api1.8.2/">BioJava legacy project</a></span></li>
<li id="cite_note-fatcat-20"><span class="mw-cite-backlink">^ <a href="#cite_ref-fatcat_20-0"><sup><i><b>a</b></i></sup></a> <a href="#cite_ref-fatcat_20-1"><sup><i><b>b</b></i></sup></a> <a href="#cite_ref-fatcat_20-2"><sup><i><b>c</b></i></sup></a></span> <span class="reference-text"><cite class="citation journal">Ye Y, Godzik A (October 2003). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=14534198">"Flexible structure alignment by chaining aligned fragment pairs allowing twists"</a>. <i>Bioinformatics</i>. <b>19</b> (Suppl 2): ii246–55. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbtg1086">10.1093/bioinformatics/btg1086</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/14534198">14534198</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Flexible+structure+alignment+by+chaining+aligned+fragment+pairs+allowing+twists&amp;rft.aufirst=Y&amp;rft.au=Godzik%2C+A&amp;rft.aulast=Ye&amp;rft.date=2003-10&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D14534198&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbtg1086&amp;rft_id=info%3Apmid%2F14534198&amp;rft.issue=Suppl+2&amp;rft.jtitle=Bioinformatics&amp;rft.pages=ii246-55&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=19" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-21"><span class="mw-cite-backlink"><b><a href="#cite_ref-21">^</a></b></span> <span class="reference-text"><cite class="citation journal">Shindyalov IN, Bourne PE (September 1998). <a rel="nofollow" class="external text" href="http://peds.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=9796821">"Protein structure alignment by incremental combinatorial extension (CE) of the optimal path"</a>. <i>Protein Eng</i>. <b>11</b> (9): 739–47. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fprotein%2F11.9.739">10.1093/protein/11.9.739</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/9796821">9796821</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Protein+structure+alignment+by+incremental+combinatorial+extension+%28CE%29+of+the+optimal+path&amp;rft.au=Bourne%2C+PE&amp;rft.aufirst=IN&amp;rft.aulast=Shindyalov&amp;rft.date=1998-09&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fpeds.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D9796821&amp;rft_id=info%3Adoi%2F10.1093%2Fprotein%2F11.9.739&amp;rft_id=info%3Apmid%2F9796821&amp;rft.issue=9&amp;rft.jtitle=Protein+Eng.&amp;rft.pages=739-47&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=11" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-22"><span class="mw-cite-backlink"><b><a href="#cite_ref-22">^</a></b></span> <span class="reference-text"><cite class="citation journal">Bliven S, Prlić A (2012). <a rel="nofollow" class="external text" href="http://dx.plos.org/10.1371/journal.pcbi.1002445">"Circular permutation in proteins"</a>. <i>PLoS Comput. Biol</i>. <b>8</b> (3): e1002445. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1371%2Fjournal.pcbi.1002445">10.1371/journal.pcbi.1002445</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3320104">3320104</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/22496628">22496628</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Circular+permutation+in+proteins&amp;rft.aufirst=S&amp;rft.aulast=Bliven&amp;rft.au=Prli%C4%87%2C+A&amp;rft.date=2012&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3320104&amp;rft_id=http%3A%2F%2Fdx.plos.org%2F10.1371%2Fjournal.pcbi.1002445&amp;rft_id=info%3Adoi%2F10.1371%2Fjournal.pcbi.1002445&amp;rft_id=info%3Apmid%2F22496628&amp;rft.issue=3&amp;rft.jtitle=PLoS+Comput.+Biol.&amp;rft.pages=e1002445&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=8" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-23"><span class="mw-cite-backlink"><b><a href="#cite_ref-23">^</a></b></span> <span class="reference-text"><cite class="citation journal">Rose PW, Beran B, Bi C, et al. (January 2011). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=21036868">"The RCSB Protein Data Bank: redesigned web site and web services"</a>. <i>Nucleic Acids Res</i>. <b>39</b> (Database issue): D392–401. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2Fgkq1021">10.1093/nar/gkq1021</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3013649">3013649</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/21036868">21036868</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+RCSB+Protein+Data+Bank%3A+redesigned+web+site+and+web+services&amp;rft.au=Beran%2C+B&amp;rft.au=Bi%2C+C&amp;rft.aufirst=PW&amp;rft.aulast=Rose&amp;rft.date=2011-01&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3013649&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D21036868&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkq1021&amp;rft_id=info%3Apmid%2F21036868&amp;rft.issue=Database+issue&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=D392-401&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=39" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-24"><span class="mw-cite-backlink"><b><a href="#cite_ref-24">^</a></b></span> <span class="reference-text"><cite class="citation journal">Prlić A, Bliven S, Rose PW, et al. (December 2010). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=20937596">"Pre-calculated protein structure alignments at the RCSB PDB website"</a>. <i>Bioinformatics</i>. <b>26</b> (23): 2983–5. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbtq572">10.1093/bioinformatics/btq572</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3003546">3003546</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/20937596">20937596</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Pre-calculated+protein+structure+alignments+at+the+RCSB+PDB+website&amp;rft.au=Bliven%2C+S&amp;rft.aufirst=A&amp;rft.aulast=Prli%C4%87&amp;rft.au=Rose%2C+PW&amp;rft.date=2010-12&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3003546&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D20937596&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbtq572&amp;rft_id=info%3Apmid%2F20937596&amp;rft.issue=23&amp;rft.jtitle=Bioinformatics&amp;rft.pages=2983-5&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=26" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-25"><span class="mw-cite-backlink"><b><a href="#cite_ref-25">^</a></b></span> <span class="reference-text"><cite class="citation journal">Bernstein FC, Koetzle TF, Williams GJ, et al. (May 1977). "The Protein Data Bank: a computer-based archival file for macromolecular structures". <i>J. Mol. Biol</i>. <b>112</b> (3): 535–42. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1016%2Fs0022-2836(77)80200-3">10.1016/s0022-2836(77)80200-3</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/875032">875032</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+Protein+Data+Bank%3A+a+computer-based+archival+file+for+macromolecular+structures&amp;rft.aufirst=FC&amp;rft.au=Koetzle%2C+TF&amp;rft.aulast=Bernstein&amp;rft.au=Williams%2C+GJ&amp;rft.date=1977-05&amp;rft.genre=article&amp;rft_id=info%3Adoi%2F10.1016%2Fs0022-2836%2877%2980200-3&amp;rft_id=info%3Apmid%2F875032&amp;rft.issue=3&amp;rft.jtitle=J.+Mol.+Biol.&amp;rft.pages=535-42&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=112" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-26"><span class="mw-cite-backlink"><b><a href="#cite_ref-26">^</a></b></span> <span class="reference-text">Fitzgerald,P.M.D. et al. (2006) Macromolecular dictionary (mmCIF). In Hall,S.R.</span></li>
<li id="cite_note-27"><span class="mw-cite-backlink"><b><a href="#cite_ref-27">^</a></b></span> <span class="reference-text"><cite class="citation journal">Velankar S, McNeil P, Mittard-Runte V, et al. (January 2005). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=15608192">"E-MSD: an integrated data resource for bioinformatics"</a>. <i>Nucleic Acids Res</i>. <b>33</b> (Database issue): D262–5. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2Fgki058">10.1093/nar/gki058</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC540012">540012</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/15608192">15608192</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=E-MSD%3A+an+integrated+data+resource+for+bioinformatics&amp;rft.aufirst=S&amp;rft.aulast=Velankar&amp;rft.au=McNeil%2C+P&amp;rft.au=Mittard-Runte%2C+V&amp;rft.date=2005-01&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC540012&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D15608192&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgki058&amp;rft_id=info%3Apmid%2F15608192&amp;rft.issue=Database+issue&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=D262-5&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=33" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-28"><span class="mw-cite-backlink"><b><a href="#cite_ref-28">^</a></b></span> <span class="reference-text"><cite class="citation journal">Moreland JL, Gramada A, Buzko OV, Zhang Q, Bourne PE (2005). <a rel="nofollow" class="external text" href="http://www.biomedcentral.com/1471-2105/6/21">"The Molecular Biology Toolkit (MBT): a modular platform for developing molecular visualization applications"</a>. <i>BMC Bioinformatics</i>. <b>6</b>: 21. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1186%2F1471-2105-6-21">10.1186/1471-2105-6-21</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC548701">548701</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/15694009">15694009</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+Molecular+Biology+Toolkit+%28MBT%29%3A+a+modular+platform+for+developing+molecular+visualization+applications&amp;rft.au=Bourne%2C+PE&amp;rft.au=Buzko%2C+OV&amp;rft.aufirst=JL&amp;rft.au=Gramada%2C+A&amp;rft.aulast=Moreland&amp;rft.au=Zhang%2C+Q&amp;rft.date=2005&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC548701&amp;rft_id=http%3A%2F%2Fwww.biomedcentral.com%2F1471-2105%2F6%2F21&amp;rft_id=info%3Adoi%2F10.1186%2F1471-2105-6-21&amp;rft_id=info%3Apmid%2F15694009&amp;rft.jtitle=BMC+Bioinformatics&amp;rft.pages=21&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=6" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-29"><span class="mw-cite-backlink"><b><a href="#cite_ref-29">^</a></b></span> <span class="reference-text"><cite class="citation journal">Besemer J, Borodovsky M (July 2005). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=15980510">"GeneMark: web software for gene finding in prokaryotes, eukaryotes and viruses"</a>. <i>Nucleic Acids Res</i>. <b>33</b> (Web Server issue): W451–4. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2Fgki487">10.1093/nar/gki487</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1160247">1160247</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/15980510">15980510</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=GeneMark%3A+web+software+for+gene+finding+in+prokaryotes%2C+eukaryotes+and+viruses&amp;rft.au=Borodovsky%2C+M&amp;rft.aufirst=J&amp;rft.aulast=Besemer&amp;rft.date=2005-07&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC1160247&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D15980510&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgki487&amp;rft_id=info%3Apmid%2F15980510&amp;rft.issue=Web+Server+issue&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=W451-4&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=33" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-30"><span class="mw-cite-backlink"><b><a href="#cite_ref-30">^</a></b></span> <span class="reference-text"><cite class="citation journal">Blanco E, Abril JF (2009). "Computational gene annotation in new genome assemblies using GeneID". <i>Methods Mol. Biol</i>. <b>537</b>: 243–61. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1007%2F978-1-59745-251-9_12">10.1007/978-1-59745-251-9_12</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/19378148">19378148</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Computational+gene+annotation+in+new+genome+assemblies+using+GeneID&amp;rft.au=Abril%2C+JF&amp;rft.aufirst=E&amp;rft.aulast=Blanco&amp;rft.date=2009&amp;rft.genre=article&amp;rft_id=info%3Adoi%2F10.1007%2F978-1-59745-251-9_12&amp;rft_id=info%3Apmid%2F19378148&amp;rft.jtitle=Methods+Mol.+Biol.&amp;rft.pages=243-61&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=537" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-31"><span class="mw-cite-backlink"><b><a href="#cite_ref-31">^</a></b></span> <span class="reference-text"><cite class="citation journal">Kelley DR, Liu B, Delcher AL, Pop M, Salzberg SL (January 2012). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=22102569">"Gene prediction with Glimmer for metagenomic sequences augmented by classification and clustering"</a>. <i>Nucleic Acids Res</i>. <b>40</b> (1): e9. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2Fgkr1067">10.1093/nar/gkr1067</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245904">3245904</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/22102569">22102569</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Gene+prediction+with+Glimmer+for+metagenomic+sequences+augmented+by+classification+and+clustering&amp;rft.au=Delcher%2C+AL&amp;rft.aufirst=DR&amp;rft.aulast=Kelley&amp;rft.au=Liu%2C+B&amp;rft.au=Pop%2C+M&amp;rft.au=Salzberg%2C+SL&amp;rft.date=2012-01&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3245904&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D22102569&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkr1067&amp;rft_id=info%3Apmid%2F22102569&amp;rft.issue=1&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=e9&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=40" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-32"><span class="mw-cite-backlink"><b><a href="#cite_ref-32">^</a></b></span> <span class="reference-text"><cite class="citation journal">Stein LD, Mungall C, Shu S, et al. (October 2002). <a rel="nofollow" class="external text" href="http://genome.cshlp.org/cgi/pmidlookup?view=long&amp;pmid=12368253">"The generic genome browser: a building block for a model organism system database"</a>. <i>Genome Res</i>. <b>12</b> (10): 1599–610. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1101%2Fgr.403602">10.1101/gr.403602</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC187535">187535</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/12368253">12368253</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+generic+genome+browser%3A+a+building+block+for+a+model+organism+system+database&amp;rft.aufirst=LD&amp;rft.aulast=Stein&amp;rft.au=Mungall%2C+C&amp;rft.au=Shu%2C+S&amp;rft.date=2002-10&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC187535&amp;rft_id=http%3A%2F%2Fgenome.cshlp.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D12368253&amp;rft_id=info%3Adoi%2F10.1101%2Fgr.403602&amp;rft_id=info%3Apmid%2F12368253&amp;rft.issue=10&amp;rft.jtitle=Genome+Res.&amp;rft.pages=1599-610&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=12" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-33"><span class="mw-cite-backlink"><b><a href="#cite_ref-33">^</a></b></span> <span class="reference-text"><cite class="citation journal">Cock PJ, Fields CJ, Goto N, Heuer ML, Rice PM (April 2010). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=20015970">"The Sanger FASTQ file format for sequences with quality scores, and the Solexa/Illumina FASTQ variants"</a>. <i>Nucleic Acids Res</i>. <b>38</b> (6): 1767–71. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2Fgkp1137">10.1093/nar/gkp1137</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2847217">2847217</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/20015970">20015970</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+Sanger+FASTQ+file+format+for+sequences+with+quality+scores%2C+and+the+Solexa%2FIllumina+FASTQ+variants&amp;rft.au=Fields%2C+CJ&amp;rft.aufirst=PJ&amp;rft.au=Goto%2C+N&amp;rft.au=Heuer%2C+ML&amp;rft.aulast=Cock&amp;rft.au=Rice%2C+PM&amp;rft.date=2010-04&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC2847217&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D20015970&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkp1137&amp;rft_id=info%3Apmid%2F20015970&amp;rft.issue=6&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=1767-71&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=38" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-34"><span class="mw-cite-backlink"><b><a href="#cite_ref-34">^</a></b></span> <span class="reference-text"><cite class="citation journal">Needleman SB, Wunsch CD (March 1970). <a rel="nofollow" class="external text" href="http://linkinghub.elsevier.com/retrieve/pii/0022-2836(70)90057-4">"A general method applicable to the search for similarities in the amino acid sequence of two proteins"</a>. <i>J. Mol. Biol</i>. <b>48</b> (3): 443–53. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1016%2F0022-2836(70)90057-4">10.1016/0022-2836(70)90057-4</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/5420325">5420325</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=A+general+method+applicable+to+the+search+for+similarities+in+the+amino+acid+sequence+of+two+proteins&amp;rft.aufirst=SB&amp;rft.aulast=Needleman&amp;rft.au=Wunsch%2C+CD&amp;rft.date=1970-03&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2F0022-2836%2870%2990057-4&amp;rft_id=info%3Adoi%2F10.1016%2F0022-2836%2870%2990057-4&amp;rft_id=info%3Apmid%2F5420325&amp;rft.issue=3&amp;rft.jtitle=J.+Mol.+Biol.&amp;rft.pages=443-53&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=48" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-35"><span class="mw-cite-backlink"><b><a href="#cite_ref-35">^</a></b></span> <span class="reference-text"><cite class="citation journal">Smith TF, Waterman MS (March 1981). <a rel="nofollow" class="external text" href="http://linkinghub.elsevier.com/retrieve/pii/0022-2836(81)90087-5">"Identification of common molecular subsequences"</a>. <i>J. Mol. Biol</i>. <b>147</b> (1): 195–7. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1016%2F0022-2836(81)90087-5">10.1016/0022-2836(81)90087-5</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/7265238">7265238</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Identification+of+common+molecular+subsequences&amp;rft.aufirst=TF&amp;rft.aulast=Smith&amp;rft.au=Waterman%2C+MS&amp;rft.date=1981-03&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2F0022-2836%2881%2990087-5&amp;rft_id=info%3Adoi%2F10.1016%2F0022-2836%2881%2990087-5&amp;rft_id=info%3Apmid%2F7265238&amp;rft.issue=1&amp;rft.jtitle=J.+Mol.+Biol.&amp;rft.pages=195-7&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=147" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-36"><span class="mw-cite-backlink"><b><a href="#cite_ref-36">^</a></b></span> <span class="reference-text"><cite class="citation journal">Guan X, Uberbacher EC (February 1996). "Alignments of DNA and protein sequences containing frameshift errors". <i>Comput. Appl. Biosci</i>. <b>12</b> (1): 31–40. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2F12.1.31">10.1093/bioinformatics/12.1.31</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/8670617">8670617</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Alignments+of+DNA+and+protein+sequences+containing+frameshift+errors&amp;rft.aufirst=X&amp;rft.aulast=Guan&amp;rft.au=Uberbacher%2C+EC&amp;rft.date=1996-02&amp;rft.genre=article&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2F12.1.31&amp;rft_id=info%3Apmid%2F8670617&amp;rft.issue=1&amp;rft.jtitle=Comput.+Appl.+Biosci.&amp;rft.pages=31-40&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=12" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-37"><span class="mw-cite-backlink"><b><a href="#cite_ref-37">^</a></b></span> <span class="reference-text"><cite class="citation journal">Chen K, Jung YS, Bonagura CA, et al. (February 2002). <a rel="nofollow" class="external text" href="http://www.jbc.org/cgi/pmidlookup?view=long&amp;pmid=11704670">"Azotobacter vinelandii ferredoxin I: a sequence and structure comparison approach to alteration of [4Fe-4S]2+/+ reduction potential"</a>. <i>J. Biol. Chem</i>. <b>277</b> (7): 5603–10. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1074%2Fjbc.M108916200">10.1074/jbc.M108916200</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/11704670">11704670</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Azotobacter+vinelandii+ferredoxin+I%3A+a+sequence+and+structure+comparison+approach+to+alteration+of+%5B4Fe-4S%5D2%2B%2F%2B+reduction+potential&amp;rft.au=Bonagura%2C+CA&amp;rft.aufirst=K&amp;rft.au=Jung%2C+YS&amp;rft.aulast=Chen&amp;rft.date=2002-02&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fwww.jbc.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D11704670&amp;rft_id=info%3Adoi%2F10.1074%2Fjbc.M108916200&amp;rft_id=info%3Apmid%2F11704670&amp;rft.issue=7&amp;rft.jtitle=J.+Biol.+Chem.&amp;rft.pages=5603-10&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=277" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-38"><span class="mw-cite-backlink"><b><a href="#cite_ref-38">^</a></b></span> <span class="reference-text"><cite class="citation journal">Montecchi-Palazzi L, Beavis R, Binz PA, et al. (August 2008). "The PSI-MOD community standard for representation of protein modification data". <i>Nat. Biotechnol</i>. <b>26</b> (8): 864–6. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1038%2Fnbt0808-864">10.1038/nbt0808-864</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/18688235">18688235</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+PSI-MOD+community+standard+for+representation+of+protein+modification+data&amp;rft.au=Beavis%2C+R&amp;rft.au=Binz%2C+PA&amp;rft.aufirst=L&amp;rft.aulast=Montecchi-Palazzi&amp;rft.date=2008-08&amp;rft.genre=article&amp;rft_id=info%3Adoi%2F10.1038%2Fnbt0808-864&amp;rft_id=info%3Apmid%2F18688235&amp;rft.issue=8&amp;rft.jtitle=Nat.+Biotechnol.&amp;rft.pages=864-6&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=26" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-39"><span class="mw-cite-backlink"><b><a href="#cite_ref-39">^</a></b></span> <span class="reference-text"><cite class="citation journal">Garavelli JS (June 2004). "The RESID Database of Protein Modifications as a resource and annotation tool". <i>Proteomics</i>. <b>4</b> (6): 1527–33. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1002%2Fpmic.200300777">10.1002/pmic.200300777</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/15174122">15174122</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+RESID+Database+of+Protein+Modifications+as+a+resource+and+annotation+tool&amp;rft.au=Garavelli+JS&amp;rft.date=2004-06&amp;rft.genre=article&amp;rft_id=info%3Adoi%2F10.1002%2Fpmic.200300777&amp;rft_id=info%3Apmid%2F15174122&amp;rft.issue=6&amp;rft.jtitle=Proteomics&amp;rft.pages=1527-33&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=4" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-40"><span class="mw-cite-backlink"><b><a href="#cite_ref-40">^</a></b></span> <span class="reference-text"><cite class="citation journal">Berman HM, Westbrook J, Feng Z, et al. (January 2000). <a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC102472">"The Protein Data Bank"</a>. <i>Nucleic Acids Res</i>. <b>28</b> (1): 235–42. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2F28.1.235">10.1093/nar/28.1.235</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC102472">102472</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/10592235">10592235</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=The+Protein+Data+Bank&amp;rft.au=Feng%2C+Z&amp;rft.aufirst=HM&amp;rft.aulast=Berman&amp;rft.au=Westbrook%2C+J&amp;rft.date=2000-01&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC102472&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC102472&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2F28.1.235&amp;rft_id=info%3Apmid%2F10592235&amp;rft.issue=1&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=235-42&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=28" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-41"><span class="mw-cite-backlink"><b><a href="#cite_ref-41">^</a></b></span> <span class="reference-text"><cite class="citation web">CookBook3, BioJava. <a rel="nofollow" class="external text" href="http://biojava.org/wiki/BioJava:CookBook3:ModFinder">"ModFinder"</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.aufirst=BioJava&amp;rft.aulast=CookBook3&amp;rft.btitle=ModFinder&amp;rft.genre=unknown&amp;rft_id=http%3A%2F%2Fbiojava.org%2Fwiki%2FBioJava%3ACookBook3%3AModFinder&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-42"><span class="mw-cite-backlink"><b><a href="#cite_ref-42">^</a></b></span> <span class="reference-text"><cite class="citation journal">Farriol-Mathis N, Garavelli JS, Boeckmann B, et al. (June 2004). "Annotation of post-translational modifications in the Swiss-Prot knowledge base". <i>Proteomics</i>. <b>4</b> (6): 1537–50. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1002%2Fpmic.200300764">10.1002/pmic.200300764</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/15174124">15174124</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=Annotation+of+post-translational+modifications+in+the+Swiss-Prot+knowledge+base&amp;rft.au=Boeckmann%2C+B&amp;rft.aufirst=N&amp;rft.au=Garavelli%2C+JS&amp;rft.aulast=Farriol-Mathis&amp;rft.date=2004-06&amp;rft.genre=article&amp;rft_id=info%3Adoi%2F10.1002%2Fpmic.200300764&amp;rft_id=info%3Apmid%2F15174124&amp;rft.issue=6&amp;rft.jtitle=Proteomics&amp;rft.pages=1537-50&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=4" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-43"><span class="mw-cite-backlink"><b><a href="#cite_ref-43">^</a></b></span> <span class="reference-text"><cite class="citation journal">Yang ZR, Thomson R, McNeil P, Esnouf RM (August 2005). <a rel="nofollow" class="external text" href="http://bioinformatics.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=15947016">"RONN: the bio-basis function neural network technique applied to the detection of natively disordered regions in proteins"</a>. <i>Bioinformatics</i>. <b>21</b> (16): 3369–76. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fbioinformatics%2Fbti534">10.1093/bioinformatics/bti534</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/15947016">15947016</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=RONN%3A+the+bio-basis+function+neural+network+technique+applied+to+the+detection+of+natively+disordered+regions+in+proteins&amp;rft.au=Esnouf%2C+RM&amp;rft.aufirst=ZR&amp;rft.aulast=Yang&amp;rft.au=McNeil%2C+P&amp;rft.au=Thomson%2C+R&amp;rft.date=2005-08&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fbioinformatics.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D15947016&amp;rft_id=info%3Adoi%2F10.1093%2Fbioinformatics%2Fbti534&amp;rft_id=info%3Apmid%2F15947016&amp;rft.issue=16&amp;rft.jtitle=Bioinformatics&amp;rft.pages=3369-76&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=21" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-44"><span class="mw-cite-backlink"><b><a href="#cite_ref-44">^</a></b></span> <span class="reference-text"><cite class="citation web">CookBook3, BioJava. <a rel="nofollow" class="external text" href="http://biojava.org/wiki/BioJava:CookBook3:ProteinDisorderCLI">"Protein Disorder CLI"</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.aufirst=BioJava&amp;rft.aulast=CookBook3&amp;rft.btitle=Protein+Disorder+CLI&amp;rft.genre=unknown&amp;rft_id=http%3A%2F%2Fbiojava.org%2Fwiki%2FBioJava%3ACookBook3%3AProteinDisorderCLI&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-45"><span class="mw-cite-backlink"><b><a href="#cite_ref-45">^</a></b></span> <span class="reference-text"><cite class="citation journal">Finn RD, Clements J, Eddy SR (July 2011). <a rel="nofollow" class="external text" href="http://nar.oxfordjournals.org/cgi/pmidlookup?view=long&amp;pmid=21593126">"HMMER web server: interactive sequence similarity searching"</a>. <i>Nucleic Acids Res</i>. <b>39</b> (Web Server issue): W29–37. <a href="https://en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1093%2Fnar%2Fgkr367">10.1093/nar/gkr367</a>. <a href="https://en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<span class="plainlinks"><a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773">3125773</a><span style="margin-left:0.1em"><img alt="Freely accessible" src="http://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/9px-Lock-green.svg.png" title="Freely accessible" width="9" height="14" srcset="//upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/14px-Lock-green.svg.png 1.5x, //upload.wikimedia.org/wikipedia/commons/thumb/6/65/Lock-green.svg/18px-Lock-green.svg.png 2x" data-file-width="512" data-file-height="813" /></span></span>. <a href="https://en.wikipedia.org/wiki/PubMed_Identifier" class="mw-redirect" title="PubMed Identifier">PMID</a>&#160;<a rel="nofollow" class="external text" href="http://www.ncbi.nlm.nih.gov/pubmed/21593126">21593126</a>.</cite><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ABioJava&amp;rft.atitle=HMMER+web+server%3A+interactive+sequence+similarity+searching&amp;rft.au=Clements%2C+J&amp;rft.au=Eddy%2C+SR&amp;rft.aufirst=RD&amp;rft.aulast=Finn&amp;rft.date=2011-07&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3125773&amp;rft_id=http%3A%2F%2Fnar.oxfordjournals.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D21593126&amp;rft_id=info%3Adoi%2F10.1093%2Fnar%2Fgkr367&amp;rft_id=info%3Apmid%2F21593126&amp;rft.issue=Web+Server+issue&amp;rft.jtitle=Nucleic+Acids+Res.&amp;rft.pages=W29-37&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=39" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
</ol>
</div>


<!-- 
NewPP limit report
Parsed by mw1182
Cached time: 20170117104937
Cache expiry: 86400
Dynamic content: true
CPU time usage: 0.428 seconds
Real time usage: 0.481 seconds
Preprocessor visited node count: 2853/1000000
Preprocessor generated node count: 0/1500000
Post‐expand include size: 106838/2097152 bytes
Template argument size: 2212/2097152 bytes
Highest expansion depth: 25/40
Expensive parser function count: 3/500
Lua time usage: 0.236/10.000 seconds
Lua memory usage: 4.49 MB/50 MB
-->

<!-- 
Transclusion expansion time report (%,ms,calls,template)
100.00%  424.741      1 - -total
 64.03%  271.947      1 - Template:Reflist
 47.19%  200.420     35 - Template:Cite_journal
 20.03%   85.083      2 - Template:Infobox
 15.09%   64.079      1 - Template:Infobox_software
 10.03%   42.597      1 - Template:Howto
  6.66%   28.269      1 - Template:Infobox_software/simple
  6.60%   28.045      1 - Template:Ambox
  5.12%   21.732      3 - Template:Start_date_and_age
  3.46%   14.701      1 - Template:Official_website
-->

<!-- Saved in parser cache with key enwiki:pcache:idhash:559858-0!*!0!!en!4!* and timestamp 20170117104936 and revision id 755935617
 -->
<noscript><img src="http://en.wikipedia.org/wiki/Special:CentralAutoLogin/start?type=1x1" alt="" title="" width="1" height="1" style="border: none; position: absolute;" /></noscript></div>					<div class="printfooter">
						Retrieved from "<a dir="ltr" href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;oldid=755935617">https://en.wikipedia.org/w/index.php?title=BioJava&amp;oldid=755935617</a>"					</div>
				<div id="catlinks" class="catlinks" data-mw="interface"><div id="mw-normal-catlinks" class="mw-normal-catlinks"><a href="Help_Category.html" title="Help:Category">Categories</a>: <ul><li><a href="https://en.wikipedia.org/wiki/Category:Bioinformatics" title="Category:Bioinformatics">Bioinformatics</a></li><li><a href="https://en.wikipedia.org/wiki/Category:Bioinformatics_software" title="Category:Bioinformatics software">Bioinformatics software</a></li><li><a href="Category_Java_platform_software.html" title="Category:Java platform software">Java platform software</a></li></ul></div><div id="mw-hidden-catlinks" class="mw-hidden-catlinks mw-hidden-cats-hidden">Hidden categories: <ul><li><a href="https://en.wikipedia.org/wiki/Category:Articles_needing_cleanup_from_July_2016" title="Category:Articles needing cleanup from July 2016">Articles needing cleanup from July 2016</a></li><li><a href="https://en.wikipedia.org/wiki/Category:All_pages_needing_cleanup" title="Category:All pages needing cleanup">All pages needing cleanup</a></li><li><a href="https://en.wikipedia.org/wiki/Category:Articles_containing_how-to_sections" title="Category:Articles containing how-to sections">Articles containing how-to sections</a></li><li><a href="https://en.wikipedia.org/wiki/Category:Official_website_not_in_Wikidata" title="Category:Official website not in Wikidata">Official website not in Wikidata</a></li></ul></div></div>				<div class="visualClear"></div>
							</div>
		</div>
		<div id="mw-navigation">
			<h2>Navigation menu</h2>

			<div id="mw-head">
									<div id="p-personal" role="navigation" class="" aria-labelledby="p-personal-label">
						<h3 id="p-personal-label">Personal tools</h3>
						<ul>
							<li id="pt-anonuserpage">Not logged in</li><li id="pt-anontalk"><a href="Special_MyTalk.html" title="Discussion about edits from this IP address [n]" accesskey="n">Talk</a></li><li id="pt-anoncontribs"><a href="Special_MyContributions.html" title="A list of edits made from this IP address [y]" accesskey="y">Contributions</a></li><li id="pt-createaccount"><a href="https://en.wikipedia.org/w/index.php?title=Special:CreateAccount&amp;returnto=BioJava" title="You are encouraged to create an account and log in; however, it is not mandatory">Create account</a></li><li id="pt-login"><a href="https://en.wikipedia.org/w/index.php?title=Special:UserLogin&amp;returnto=BioJava" title="You're encouraged to log in; however, it's not mandatory. [o]" accesskey="o">Log in</a></li>						</ul>
					</div>
									<div id="left-navigation">
										<div id="p-namespaces" role="navigation" class="vectorTabs" aria-labelledby="p-namespaces-label">
						<h3 id="p-namespaces-label">Namespaces</h3>
						<ul>
															<li  id="ca-nstab-main" class="selected"><span><a href="BioJava.html"  title="View the content page [c]" accesskey="c">Article</a></span></li>
															<li  id="ca-talk"><span><a href="https://en.wikipedia.org/wiki/Talk:BioJava"  title="Discussion about the content page [t]" accesskey="t" rel="discussion">Talk</a></span></li>
													</ul>
					</div>
										<div id="p-variants" role="navigation" class="vectorMenu emptyPortlet" aria-labelledby="p-variants-label">
												<h3 id="p-variants-label">
							<span>Variants</span><a href="#"></a>
						</h3>

						<div class="menu">
							<ul>
															</ul>
						</div>
					</div>
									</div>
				<div id="right-navigation">
										<div id="p-views" role="navigation" class="vectorTabs" aria-labelledby="p-views-label">
						<h3 id="p-views-label">Views</h3>
						<ul>
															<li id="ca-view" class="selected"><span><a href="BioJava.html" >Read</a></span></li>
															<li id="ca-edit"><span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=edit"  title="Edit this page [e]" accesskey="e">Edit</a></span></li>
															<li id="ca-history" class="collapsible"><span><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=history"  title="Past revisions of this page [h]" accesskey="h">View history</a></span></li>
													</ul>
					</div>
										<div id="p-cactions" role="navigation" class="vectorMenu emptyPortlet" aria-labelledby="p-cactions-label">
						<h3 id="p-cactions-label"><span>More</span><a href="#"></a></h3>

						<div class="menu">
							<ul>
															</ul>
						</div>
					</div>
										<div id="p-search" role="search">
						<h3>
							<label for="searchInput">Search</label>
						</h3>

						<form action="https://en.wikipedia.org/w/index.php" id="searchform">
							<div id="simpleSearch">
							<input type="search" name="search" placeholder="Search Wikipedia" title="Search Wikipedia [f]" accesskey="f" id="searchInput"/><input type="hidden" value="Special:Search" name="title"/><input type="submit" name="fulltext" value="Search" title="Search Wikipedia for this text" id="mw-searchButton" class="searchButton mw-fallbackSearchButton"/><input type="submit" name="go" value="Go" title="Go to a page with this exact name if it exists" id="searchButton" class="searchButton"/>							</div>
						</form>
					</div>
									</div>
			</div>
			<div id="mw-panel">
				<div id="p-logo" role="banner"><a class="mw-wiki-logo" href="Main_Page.html"  title="Visit the main page"></a></div>
						<div class="portal" role="navigation" id='p-navigation' aria-labelledby='p-navigation-label'>
			<h3 id='p-navigation-label'>Navigation</h3>

			<div class="body">
									<ul>
						<li id="n-mainpage-description"><a href="Main_Page.html" title="Visit the main page [z]" accesskey="z">Main page</a></li><li id="n-contents"><a href="Portal_Contents.html" title="Guides to browsing Wikipedia">Contents</a></li><li id="n-featuredcontent"><a href="Portal_Featured_content.html" title="Featured content – the best of Wikipedia">Featured content</a></li><li id="n-currentevents"><a href="Portal_Current_events.html" title="Find background information on current events">Current events</a></li><li id="n-randompage"><a href="https://en.wikipedia.org/wiki/Special:Random" title="Load a random article [x]" accesskey="x">Random article</a></li><li id="n-sitesupport"><a href="https://donate.wikimedia.org/wiki/Special:FundraiserRedirector?utm_source=donate&amp;utm_medium=sidebar&amp;utm_campaign=C13_en.wikipedia.org&amp;uselang=en" title="Support us">Donate to Wikipedia</a></li><li id="n-shoplink"><a href="../../shop.wikimedia.org/index.html" title="Visit the Wikipedia store">Wikipedia store</a></li>					</ul>
							</div>
		</div>
			<div class="portal" role="navigation" id='p-interaction' aria-labelledby='p-interaction-label'>
			<h3 id='p-interaction-label'>Interaction</h3>

			<div class="body">
									<ul>
						<li id="n-help"><a href="Help_Contents.html" title="Guidance on how to use and edit Wikipedia">Help</a></li><li id="n-aboutsite"><a href="Wikipedia_About.html" title="Find out about Wikipedia">About Wikipedia</a></li><li id="n-portal"><a href="Wikipedia_Community_portal.html" title="About the project, what you can do, where to find things">Community portal</a></li><li id="n-recentchanges"><a href="Special_RecentChanges.html" title="A list of recent changes in the wiki [r]" accesskey="r">Recent changes</a></li><li id="n-contactpage"><a href="Wikipedia_Contact_us.html" title="How to contact Wikipedia">Contact page</a></li>					</ul>
							</div>
		</div>
			<div class="portal" role="navigation" id='p-tb' aria-labelledby='p-tb-label'>
			<h3 id='p-tb-label'>Tools</h3>

			<div class="body">
									<ul>
						<li id="t-whatlinkshere"><a href="https://en.wikipedia.org/wiki/Special:WhatLinksHere/BioJava" title="List of all English Wikipedia pages containing links to this page [j]" accesskey="j">What links here</a></li><li id="t-recentchangeslinked"><a href="https://en.wikipedia.org/wiki/Special:RecentChangesLinked/BioJava" rel="nofollow" title="Recent changes in pages linked from this page [k]" accesskey="k">Related changes</a></li><li id="t-upload"><a href="Wikipedia_File_Upload_Wizard.html" title="Upload files [u]" accesskey="u">Upload file</a></li><li id="t-specialpages"><a href="Special_SpecialPages.html" title="A list of all special pages [q]" accesskey="q">Special pages</a></li><li id="t-permalink"><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;oldid=755935617" title="Permanent link to this revision of the page">Permanent link</a></li><li id="t-info"><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;action=info" title="More information about this page">Page information</a></li><li id="t-wikibase"><a href="https://www.wikidata.org/wiki/Q1036510" title="Link to connected data repository item [g]" accesskey="g">Wikidata item</a></li><li id="t-cite"><a href="https://en.wikipedia.org/w/index.php?title=Special:CiteThisPage&amp;page=BioJava&amp;id=755935617" title="Information on how to cite this page">Cite this page</a></li>					</ul>
							</div>
		</div>
			<div class="portal" role="navigation" id='p-coll-print_export' aria-labelledby='p-coll-print_export-label'>
			<h3 id='p-coll-print_export-label'>Print/export</h3>

			<div class="body">
									<ul>
						<li id="coll-create_a_book"><a href="https://en.wikipedia.org/w/index.php?title=Special:Book&amp;bookcmd=book_creator&amp;referer=BioJava">Create a book</a></li><li id="coll-download-as-rdf2latex"><a href="https://en.wikipedia.org/w/index.php?title=Special:Book&amp;bookcmd=render_article&amp;arttitle=BioJava&amp;returnto=BioJava&amp;oldid=755935617&amp;writer=rdf2latex">Download as PDF</a></li><li id="t-print"><a href="https://en.wikipedia.org/w/index.php?title=BioJava&amp;printable=yes" title="Printable version of this page [p]" accesskey="p">Printable version</a></li>					</ul>
							</div>
		</div>
			<div class="portal" role="navigation" id='p-lang' aria-labelledby='p-lang-label'>
			<h3 id='p-lang-label'>Languages</h3>

			<div class="body">
									<ul>
						<li class="interlanguage-link interwiki-ar"><a href="https://ar.wikipedia.org/wiki/بايو_جافا" title="بايو جافا – Arabic" lang="ar" hreflang="ar" class="interlanguage-link-target">العربية</a></li><li class="interlanguage-link interwiki-es"><a href="https://es.wikipedia.org/wiki/BioJava" title="BioJava – Spanish" lang="es" hreflang="es" class="interlanguage-link-target">Español</a></li><li class="interlanguage-link interwiki-ko"><a href="https://ko.wikipedia.org/wiki/바이오자바" title="바이오자바 – Korean" lang="ko" hreflang="ko" class="interlanguage-link-target">한국어</a></li><li class="interlanguage-link interwiki-pt"><a href="https://pt.wikipedia.org/wiki/BioJava" title="BioJava – Portuguese" lang="pt" hreflang="pt" class="interlanguage-link-target">Português</a></li><li class="interlanguage-link interwiki-zh"><a href="https://zh.wikipedia.org/wiki/BioJava" title="BioJava – Chinese" lang="zh" hreflang="zh" class="interlanguage-link-target">中文</a></li><li class="uls-p-lang-dummy"><a href="#"></a></li>					</ul>
				<div class='after-portlet after-portlet-lang'><span class="wb-langlinks-edit wb-langlinks-link"><a href="https://www.wikidata.org/wiki/Q1036510#sitelinks-wikipedia" title="Edit interlanguage links" class="wbc-editpage">Edit links</a></span></div>			</div>
		</div>
				</div>
		</div>
		<div id="footer" role="contentinfo">
							<ul id="footer-info">
											<li id="footer-info-lastmod"> This page was last modified on 21 December 2016, at 00:41.</li>
											<li id="footer-info-copyright">Text is available under the <a rel="license" href="Wikipedia_Text_of_Creative_Commons_Attribution-ShareAlike_3.html">Creative Commons Attribution-ShareAlike License</a><a rel="license" href="../../creativecommons.org/licenses/by-sa/3.0/index.html" style="display:none;"></a>;
additional terms may apply.  By using this site, you agree to the <a href="../../wikimediafoundation.org/wiki/Terms_of_Use.html">Terms of Use</a> and <a href="../../wikimediafoundation.org/wiki/Privacy_policy.html">Privacy Policy</a>. Wikipedia® is a registered trademark of the <a href="../../www.wikimediafoundation.org/index.html">Wikimedia Foundation, Inc.</a>, a non-profit organization.</li>
									</ul>
							<ul id="footer-places">
											<li id="footer-places-privacy"><a href="../../wikimediafoundation.org/wiki/Privacy_policy.html" class="extiw" title="wmf:Privacy policy">Privacy policy</a></li>
											<li id="footer-places-about"><a href="Wikipedia_About.html" title="Wikipedia:About">About Wikipedia</a></li>
											<li id="footer-places-disclaimer"><a href="Wikipedia_General_disclaimer.html" title="Wikipedia:General disclaimer">Disclaimers</a></li>
											<li id="footer-places-contact"><a href="Wikipedia_Contact_us.html">Contact Wikipedia</a></li>
											<li id="footer-places-developers"><a href="../../www.mediawiki.org/wiki/Special_MyLanguage/How_to_contribute.html">Developers</a></li>
											<li id="footer-places-cookiestatement"><a href="../../wikimediafoundation.org/wiki/Cookie_statement.html">Cookie statement</a></li>
											<li id="footer-places-mobileview"><a href="http://en.m.wikipedia.org/w/index.php?title=BioJava&amp;mobileaction=toggle_view_mobile" class="noprint stopMobileRedirectToggle">Mobile view</a></li>
									</ul>
										<ul id="footer-icons" class="noprint">
											<li id="footer-copyrightico">
							<a href="../../www.wikimediafoundation.org/index.html"><img src="https://en.wikipedia.org/static/images/wikimedia-button.png" srcset="/static/images/wikimedia-button-1.5x.png 1.5x, /static/images/wikimedia-button-2x.png 2x" width="88" height="31" alt="Wikimedia Foundation"/></a>						</li>
											<li id="footer-poweredbyico">
							<a href="../../www.mediawiki.org/index.html"><img src="https://en.wikipedia.org/static/images/poweredby_mediawiki_88x31.png" alt="Powered by MediaWiki" srcset="/static/images/poweredby_mediawiki_132x47.png 1.5x, /static/images/poweredby_mediawiki_176x62.png 2x" width="88" height="31"/></a>						</li>
									</ul>
						<div style="clear:both"></div>
		</div>
		<script>(window.RLQ=window.RLQ||[]).push(function(){mw.config.set({"wgBackendResponseTime":76,"wgHostname":"mw1255"});});</script>
	</body>

<!-- Mirrored from en.wikipedia.org/wiki/BioJava by HTTrack Website Copier/3.x [XR&CO'2014], Tue, 17 Jan 2017 12:52:42 GMT -->
</html>
